--- a/net/minecraft/src/EntityItem.java
+++ b/net/minecraft/src/EntityItem.java
@@ -16,7 +16,7 @@
       this.field_801_a = p_i91_8_;
       this.field_605_aq = (float)(Math.random() * 360.0D);
       this.field_608_an = (double)((float)(Math.random() * (double)0.2F - (double)0.1F));
-      this.field_607_ao = (double)0.2F;
+      this.field_607_ao = 0.2F;
       this.field_606_ap = (double)((float)(Math.random() * (double)0.2F - (double)0.1F));
       this.field_9289_aT = false;
    }
@@ -36,9 +36,9 @@
       this.field_9285_at = this.field_611_ak;
       this.field_9284_au = this.field_610_al;
       this.field_9283_av = this.field_609_am;
-      this.field_607_ao -= (double)0.04F;
+      this.field_607_ao -= 0.04F;
       if (this.field_615_ag.func_599_f(MathHelper.func_1108_b(this.field_611_ak), MathHelper.func_1108_b(this.field_610_al), MathHelper.func_1108_b(this.field_609_am)) == Material.field_1331_g) {
-         this.field_607_ao = (double)0.2F;
+         this.field_607_ao = 0.2F;
          this.field_608_an = (double)((this.field_9312_bd.nextFloat() - this.field_9312_bd.nextFloat()) * 0.2F);
          this.field_606_ap = (double)((this.field_9312_bd.nextFloat() - this.field_9312_bd.nextFloat()) * 0.2F);
          this.field_615_ag.func_623_a(this, "random.fizz", 0.4F, 2.0F + this.field_9312_bd.nextFloat() * 0.4F);
@@ -56,9 +56,9 @@
          }
       }
 
-      this.field_608_an *= (double)f;
-      this.field_607_ao *= (double)0.98F;
-      this.field_606_ap *= (double)f;
+      this.field_608_an *= f;
+      this.field_607_ao *= 0.98F;
+      this.field_606_ap *= f;
       if (this.field_9298_aH) {
          this.field_607_ao *= -0.5D;
       }
@@ -123,27 +123,27 @@
 
          float f = this.field_9312_bd.nextFloat() * 0.2F + 0.1F;
          if (b0 == 0) {
-            this.field_608_an = (double)(-f);
+            this.field_608_an = -f;
          }
 
          if (b0 == 1) {
-            this.field_608_an = (double)f;
+            this.field_608_an = f;
          }
 
          if (b0 == 2) {
-            this.field_607_ao = (double)(-f);
+            this.field_607_ao = -f;
          }
 
          if (b0 == 3) {
-            this.field_607_ao = (double)f;
+            this.field_607_ao = f;
          }
 
          if (b0 == 4) {
-            this.field_606_ap = (double)(-f);
+            this.field_606_ap = -f;
          }
 
          if (b0 == 5) {
-            this.field_606_ap = (double)f;
+            this.field_606_ap = f;
          }
       }
 
