--- a/net/minecraft/src/EntityRenderer.java
+++ b/net/minecraft/src/EntityRenderer.java
@@ -50,7 +50,7 @@
 
    public void func_910_a(float p_910_1_) {
       if (this.field_1388_h.field_6322_g != null) {
-         double d0 = (double)this.field_1388_h.field_6327_b.func_727_b();
+         double d0 = this.field_1388_h.field_6327_b.func_727_b();
          this.field_1388_h.field_6305_x = this.field_1388_h.field_6322_g.func_416_a(d0, p_910_1_);
          double d1 = d0;
          Vec3D vec3d = this.field_1388_h.field_6322_g.func_427_e(p_910_1_);
@@ -73,14 +73,13 @@
          Vec3D vec3d2 = vec3d.func_1257_c(vec3d1.field_1776_a * d0, vec3d1.field_1775_b * d0, vec3d1.field_1779_c * d0);
          this.field_1385_k = null;
          float f = 1.0F;
-         List list = this.field_1388_h.field_6324_e.func_659_b(this.field_1388_h.field_6322_g, this.field_1388_h.field_6322_g.field_601_au.func_1170_a(vec3d1.field_1776_a * d0, vec3d1.field_1775_b * d0, vec3d1.field_1779_c * d0).func_1177_b((double)f, (double)f, (double)f));
+         List<Entity> list = this.field_1388_h.field_6324_e.func_659_b(this.field_1388_h.field_6322_g, this.field_1388_h.field_6322_g.field_601_au.func_1170_a(vec3d1.field_1776_a * d0, vec3d1.field_1775_b * d0, vec3d1.field_1779_c * d0).func_1177_b(f, f, f));
          double d2 = 0.0D;
 
-         for(int i = 0; i < list.size(); ++i) {
-            Entity entity = (Entity)list.get(i);
+         for (Entity entity : list) {
             if (entity.func_401_c_()) {
                float f1 = entity.func_4035_j_();
-               AxisAlignedBB axisalignedbb = entity.field_601_au.func_1177_b((double)f1, (double)f1, (double)f1);
+               AxisAlignedBB axisalignedbb = entity.field_601_au.func_1177_b( f1,  f1,  f1);
                MovingObjectPosition movingobjectposition = axisalignedbb.func_1169_a(vec3d, vec3d2);
                if (axisalignedbb.func_4150_a(vec3d)) {
                   if (0.0D < d2 || d2 == 0.0D) {
@@ -376,7 +375,7 @@
          GL11.glBindTexture(3553, this.field_1388_h.field_6315_n.func_1070_a("/terrain.png"));
          if (this.field_1388_h.field_6304_y.field_1576_i) {
             GL11.glColorMask(false, false, false, false);
-            int j = renderglobal.func_943_a(entityplayersp, 1, (double)p_4134_1_);
+            int j = renderglobal.func_943_a(entityplayersp, 1, p_4134_1_);
             GL11.glColorMask(true, true, true, true);
             if (this.field_1388_h.field_6304_y.field_1578_g) {
                if (i == 0) {
@@ -387,10 +386,10 @@
             }
 
             if (j > 0) {
-               renderglobal.func_944_a(1, (double)p_4134_1_);
+               renderglobal.func_944_a(1, p_4134_1_);
             }
          } else {
-            renderglobal.func_943_a(entityplayersp, 1, (double)p_4134_1_);
+            renderglobal.func_943_a(entityplayersp, 1, p_4134_1_);
          }
 
          GL11.glDepthMask(true);
@@ -455,7 +454,7 @@
       GL11.glClear(256);
       GL11.glMatrixMode(5889);
       GL11.glLoadIdentity();
-      GL11.glOrtho(0.0D, (double)i, (double)j, 0.0D, 1000.0D, 3000.0D);
+      GL11.glOrtho(0.0D, i, j, 0.0D, 1000.0D, 3000.0D);
       GL11.glMatrixMode(5888);
       GL11.glLoadIdentity();
       GL11.glTranslatef(0.0F, 0.0F, -2000.0F);
@@ -465,7 +464,7 @@
       World world = this.field_1388_h.field_6324_e;
       EntityPlayerSP entityplayersp = this.field_1388_h.field_6322_g;
       float f = 1.0F / (float)(4 - this.field_1388_h.field_6304_y.field_1580_e);
-      f = 1.0F - (float)Math.pow((double)f, 0.25D);
+      f = 1.0F - (float)Math.pow(f, 0.25D);
       Vec3D vec3d = world.func_4079_a(this.field_1388_h.field_6322_g, p_4137_1_);
       float f1 = (float)vec3d.field_1776_a;
       float f2 = (float)vec3d.field_1775_b;
@@ -553,9 +552,9 @@
    }
 
    private FloatBuffer func_908_a(float p_908_1_, float p_908_2_, float p_908_3_, float p_908_4_) {
-      ((Buffer)this.field_1392_d).clear();
+      this.field_1392_d.clear();
       this.field_1392_d.put(p_908_1_).put(p_908_2_).put(p_908_3_).put(p_908_4_);
-      ((Buffer)this.field_1392_d).flip();
+      this.field_1392_d.flip();
       return this.field_1392_d;
    }
 }
