--- a/net/minecraft/src/MapGenCaves.java
+++ b/net/minecraft/src/MapGenCaves.java
@@ -8,8 +8,8 @@
    }
 
    protected void func_869_a(int p_869_1_, int p_869_2_, byte[] p_869_3_, double p_869_4_, double p_869_6_, double p_869_8_, float p_869_10_, float p_869_11_, float p_869_12_, int p_869_13_, int p_869_14_, double p_869_15_) {
-      double d0 = (double)(p_869_1_ * 16 + 8);
-      double d1 = (double)(p_869_2_ * 16 + 8);
+      double d0 = p_869_1_ * 16 + 8;
+      double d1 = p_869_2_ * 16 + 8;
       float f = 0.0F;
       float f1 = 0.0F;
       Random random = new Random(this.field_1305_b.nextLong());
