--- a/net/minecraft/src/RenderPainting.java
+++ b/net/minecraft/src/RenderPainting.java
@@ -3,9 +3,14 @@
 import java.util.Random;
 import org.lwjgl.opengl.GL11;
 
-public class RenderPainting extends Render {
+public class RenderPainting extends Render<EntityPainting> {
    private Random field_199_d = new Random();
 
+   @Override
+   public void func_147_a(EntityPainting p_147_1_, double p_147_2_, double p_147_4_, double p_147_6_, float p_147_8_, float p_147_9_) {
+      func_158_a(p_147_1_, p_147_2_, p_147_4_, p_147_6_, p_147_8_, p_147_9_);
+   }
+
    public void func_158_a(EntityPainting p_158_1_, double p_158_2_, double p_158_4_, double p_158_6_, float p_158_8_, float p_158_9_) {
       this.field_199_d.setSeed(187L);
       GL11.glPushMatrix();
@@ -53,35 +58,35 @@
             Tessellator tessellator = Tessellator.field_1512_a;
             tessellator.func_977_b();
             tessellator.func_980_b(0.0F, 0.0F, -1.0F);
-            tessellator.func_983_a((double)f4, (double)f7, (double)f2, (double)f9, (double)f10);
-            tessellator.func_983_a((double)f5, (double)f7, (double)f2, (double)f8, (double)f10);
-            tessellator.func_983_a((double)f5, (double)f6, (double)f2, (double)f8, (double)f11);
-            tessellator.func_983_a((double)f4, (double)f6, (double)f2, (double)f9, (double)f11);
+            tessellator.func_983_a(f4, f7, f2, f9, f10);
+            tessellator.func_983_a(f5, f7, f2, f8, f10);
+            tessellator.func_983_a(f5, f6, f2, f8, f11);
+            tessellator.func_983_a(f4, f6, f2, f9, f11);
             tessellator.func_980_b(0.0F, 0.0F, 1.0F);
-            tessellator.func_983_a((double)f4, (double)f6, (double)f3, (double)f12, (double)f14);
-            tessellator.func_983_a((double)f5, (double)f6, (double)f3, (double)f13, (double)f14);
-            tessellator.func_983_a((double)f5, (double)f7, (double)f3, (double)f13, (double)f15);
-            tessellator.func_983_a((double)f4, (double)f7, (double)f3, (double)f12, (double)f15);
+            tessellator.func_983_a(f4, f6, f3, f12, f14);
+            tessellator.func_983_a(f5, f6, f3, f13, f14);
+            tessellator.func_983_a(f5, f7, f3, f13, f15);
+            tessellator.func_983_a(f4, f7, f3, f12, f15);
             tessellator.func_980_b(0.0F, -1.0F, 0.0F);
-            tessellator.func_983_a((double)f4, (double)f6, (double)f2, (double)f16, (double)f18);
-            tessellator.func_983_a((double)f5, (double)f6, (double)f2, (double)f17, (double)f18);
-            tessellator.func_983_a((double)f5, (double)f6, (double)f3, (double)f17, (double)f19);
-            tessellator.func_983_a((double)f4, (double)f6, (double)f3, (double)f16, (double)f19);
+            tessellator.func_983_a(f4, f6, f2, f16, f18);
+            tessellator.func_983_a(f5, f6, f2, f17, f18);
+            tessellator.func_983_a(f5, f6, f3, f17, f19);
+            tessellator.func_983_a(f4, f6, f3, f16, f19);
             tessellator.func_980_b(0.0F, 1.0F, 0.0F);
-            tessellator.func_983_a((double)f4, (double)f7, (double)f3, (double)f16, (double)f18);
-            tessellator.func_983_a((double)f5, (double)f7, (double)f3, (double)f17, (double)f18);
-            tessellator.func_983_a((double)f5, (double)f7, (double)f2, (double)f17, (double)f19);
-            tessellator.func_983_a((double)f4, (double)f7, (double)f2, (double)f16, (double)f19);
+            tessellator.func_983_a(f4, f7, f3, f16, f18);
+            tessellator.func_983_a(f5, f7, f3, f17, f18);
+            tessellator.func_983_a(f5, f7, f2, f17, f19);
+            tessellator.func_983_a(f4, f7, f2, f16, f19);
             tessellator.func_980_b(-1.0F, 0.0F, 0.0F);
-            tessellator.func_983_a((double)f4, (double)f6, (double)f3, (double)f21, (double)f22);
-            tessellator.func_983_a((double)f4, (double)f7, (double)f3, (double)f21, (double)f23);
-            tessellator.func_983_a((double)f4, (double)f7, (double)f2, (double)f20, (double)f23);
-            tessellator.func_983_a((double)f4, (double)f6, (double)f2, (double)f20, (double)f22);
+            tessellator.func_983_a(f4, f6, f3, f21, f22);
+            tessellator.func_983_a(f4, f7, f3, f21, f23);
+            tessellator.func_983_a(f4, f7, f2, f20, f23);
+            tessellator.func_983_a(f4, f6, f2, f20, f22);
             tessellator.func_980_b(1.0F, 0.0F, 0.0F);
-            tessellator.func_983_a((double)f5, (double)f6, (double)f2, (double)f21, (double)f22);
-            tessellator.func_983_a((double)f5, (double)f7, (double)f2, (double)f21, (double)f23);
-            tessellator.func_983_a((double)f5, (double)f7, (double)f3, (double)f20, (double)f23);
-            tessellator.func_983_a((double)f5, (double)f6, (double)f3, (double)f20, (double)f22);
+            tessellator.func_983_a(f5, f6, f2, f21, f22);
+            tessellator.func_983_a(f5, f7, f2, f21, f23);
+            tessellator.func_983_a(f5, f7, f3, f20, f23);
+            tessellator.func_983_a(f5, f6, f3, f20, f22);
             tessellator.func_982_a();
          }
       }
