--- a/net/minecraft/src/RenderFish.java
+++ b/net/minecraft/src/RenderFish.java
@@ -2,7 +2,13 @@
 
 import org.lwjgl.opengl.GL11;
 
-public class RenderFish extends Render {
+public class RenderFish extends Render<EntityFish> {
+
+   @Override
+   public void func_147_a(EntityFish p_147_1_, double p_147_2_, double p_147_4_, double p_147_6_, float p_147_8_, float p_147_9_) {
+      func_4011_a(p_147_1_, p_147_2_, p_147_4_, p_147_6_, p_147_8_, p_147_9_);
+   }
+
    public void func_4011_a(EntityFish p_4011_1_, double p_4011_2_, double p_4011_4_, double p_4011_6_, float p_4011_8_, float p_4011_9_) {
       GL11.glPushMatrix();
       GL11.glTranslatef((float)p_4011_2_, (float)p_4011_4_, (float)p_4011_6_);
