--- a/net/minecraft/src/SoundPool.java
+++ b/net/minecraft/src/SoundPool.java
@@ -10,13 +10,14 @@
 
 public class SoundPool {
    private Random field_1661_c = new Random();
-   private Map field_1660_d = new HashMap();
-   private List field_1659_e = new ArrayList();
+   private Map<String, List<SoundPoolEntry>> field_1660_d = new HashMap<>();
+   private List<SoundPoolEntry> field_1659_e = new ArrayList<>();
    public int field_1658_a = 0;
    public boolean field_1657_b = true;
 
    public SoundPoolEntry func_1117_a(String p_1117_1_, File p_1117_2_) {
       try {
+         String s1 = p_1117_1_;
          p_1117_1_ = p_1117_1_.substring(0, p_1117_1_.indexOf("."));
          if (this.field_1657_b) {
             while(Character.isDigit(p_1117_1_.charAt(p_1117_1_.length() - 1))) {
@@ -26,11 +27,11 @@
 
          p_1117_1_ = p_1117_1_.replaceAll("/", ".");
          if (!this.field_1660_d.containsKey(p_1117_1_)) {
-            this.field_1660_d.put(p_1117_1_, new ArrayList());
+            this.field_1660_d.put(p_1117_1_, new ArrayList<>());
          }
 
-         SoundPoolEntry soundpoolentry = new SoundPoolEntry(p_1117_1_, p_1117_2_.toURI().toURL());
-         ((List)this.field_1660_d.get(p_1117_1_)).add(soundpoolentry);
+         SoundPoolEntry soundpoolentry = new SoundPoolEntry(s1, p_1117_2_.toURI().toURL());
+         this.field_1660_d.get(p_1117_1_).add(soundpoolentry);
          this.field_1659_e.add(soundpoolentry);
          ++this.field_1658_a;
          return soundpoolentry;
@@ -41,11 +42,11 @@
    }
 
    public SoundPoolEntry func_1118_a(String p_1118_1_) {
-      List list = (List)this.field_1660_d.get(p_1118_1_);
-      return list == null ? null : (SoundPoolEntry)list.get(this.field_1661_c.nextInt(list.size()));
+      List<SoundPoolEntry> list = this.field_1660_d.get(p_1118_1_);
+      return list == null ? null : list.get(this.field_1661_c.nextInt(list.size()));
    }
 
    public SoundPoolEntry func_1116_a() {
-      return this.field_1659_e.size() == 0 ? null : (SoundPoolEntry)this.field_1659_e.get(this.field_1661_c.nextInt(this.field_1659_e.size()));
+      return this.field_1659_e.size() == 0 ? null : this.field_1659_e.get(this.field_1661_c.nextInt(this.field_1659_e.size()));
    }
 }
