--- a/net/minecraft/client/Minecraft.java
+++ b/net/minecraft/client/Minecraft.java
@@ -262,7 +262,7 @@
       GL11.glClear(16640);
       GL11.glMatrixMode(5889);
       GL11.glLoadIdentity();
-      GL11.glOrtho(0.0D, (double)i, (double)j, 0.0D, 1000.0D, 3000.0D);
+      GL11.glOrtho(0.0D, i, j, 0.0D, 1000.0D, 3000.0D);
       GL11.glMatrixMode(5888);
       GL11.glLoadIdentity();
       GL11.glTranslatef(0.0F, 0.0F, -2000.0F);
@@ -275,9 +275,9 @@
       GL11.glBindTexture(3553, this.field_6315_n.func_1070_a("/title/mojang.png"));
       tessellator.func_977_b();
       tessellator.func_990_b(16777215);
-      tessellator.func_983_a(0.0D, (double)this.field_6325_d, 0.0D, 0.0D, 0.0D);
-      tessellator.func_983_a((double)this.field_6326_c, (double)this.field_6325_d, 0.0D, 0.0D, 0.0D);
-      tessellator.func_983_a((double)this.field_6326_c, 0.0D, 0.0D, 0.0D, 0.0D);
+      tessellator.func_983_a(0.0D, this.field_6325_d, 0.0D, 0.0D, 0.0D);
+      tessellator.func_983_a(this.field_6326_c, this.field_6325_d, 0.0D, 0.0D, 0.0D);
+      tessellator.func_983_a(this.field_6326_c, 0.0D, 0.0D, 0.0D, 0.0D);
       tessellator.func_983_a(0.0D, 0.0D, 0.0D, 0.0D, 0.0D);
       tessellator.func_982_a();
       short short1 = 256;
@@ -297,10 +297,10 @@
       float f1 = 0.00390625F;
       Tessellator tessellator = Tessellator.field_1512_a;
       tessellator.func_977_b();
-      tessellator.func_983_a((double)(p_6274_1_ + 0), (double)(p_6274_2_ + p_6274_6_), 0.0D, (double)((float)(p_6274_3_ + 0) * f), (double)((float)(p_6274_4_ + p_6274_6_) * f1));
-      tessellator.func_983_a((double)(p_6274_1_ + p_6274_5_), (double)(p_6274_2_ + p_6274_6_), 0.0D, (double)((float)(p_6274_3_ + p_6274_5_) * f), (double)((float)(p_6274_4_ + p_6274_6_) * f1));
-      tessellator.func_983_a((double)(p_6274_1_ + p_6274_5_), (double)(p_6274_2_ + 0), 0.0D, (double)((float)(p_6274_3_ + p_6274_5_) * f), (double)((float)(p_6274_4_ + 0) * f1));
-      tessellator.func_983_a((double)(p_6274_1_ + 0), (double)(p_6274_2_ + 0), 0.0D, (double)((float)(p_6274_3_ + 0) * f), (double)((float)(p_6274_4_ + 0) * f1));
+      tessellator.func_983_a(p_6274_1_ + 0, p_6274_2_ + p_6274_6_, 0.0D, ((float)(p_6274_3_ + 0) * f), ((float)(p_6274_4_ + p_6274_6_) * f1));
+      tessellator.func_983_a(p_6274_1_ + p_6274_5_, p_6274_2_ + p_6274_6_, 0.0D, ((float)(p_6274_3_ + p_6274_5_) * f), ((float)(p_6274_4_ + p_6274_6_) * f1));
+      tessellator.func_983_a(p_6274_1_ + p_6274_5_, p_6274_2_ + 0, 0.0D, ((float)(p_6274_3_ + p_6274_5_) * f), ((float)(p_6274_4_ + 0) * f1));
+      tessellator.func_983_a(p_6274_1_ + 0, p_6274_2_ + 0, 0.0D, ((float)(p_6274_3_ + 0) * f), ((float)(p_6274_4_ + 0) * f1));
       tessellator.func_982_a();
    }
 
@@ -586,7 +586,7 @@
       GL11.glClear(256);
       GL11.glMatrixMode(5889);
       GL11.glLoadIdentity();
-      GL11.glOrtho(0.0D, (double)this.field_6326_c, (double)this.field_6325_d, 0.0D, 1000.0D, 3000.0D);
+      GL11.glOrtho(0.0D, this.field_6326_c, this.field_6325_d, 0.0D, 1000.0D, 3000.0D);
       GL11.glMatrixMode(5888);
       GL11.glLoadIdentity();
       GL11.glTranslatef(0.0F, 0.0F, -2000.0F);
@@ -596,29 +596,29 @@
       tessellator.func_992_a(7);
       int k = (int)(i / 200000L);
       tessellator.func_990_b(536870912);
-      tessellator.func_991_a(0.0D, (double)(this.field_6325_d - k), 0.0D);
-      tessellator.func_991_a(0.0D, (double)this.field_6325_d, 0.0D);
-      tessellator.func_991_a((double)field_9240_E.length, (double)this.field_6325_d, 0.0D);
-      tessellator.func_991_a((double)field_9240_E.length, (double)(this.field_6325_d - k), 0.0D);
+      tessellator.func_991_a(0.0D, this.field_6325_d - k, 0.0D);
+      tessellator.func_991_a(0.0D, this.field_6325_d, 0.0D);
+      tessellator.func_991_a(field_9240_E.length, this.field_6325_d, 0.0D);
+      tessellator.func_991_a(field_9240_E.length, this.field_6325_d - k, 0.0D);
       tessellator.func_990_b(538968064);
-      tessellator.func_991_a(0.0D, (double)(this.field_6325_d - k * 2), 0.0D);
-      tessellator.func_991_a(0.0D, (double)(this.field_6325_d - k), 0.0D);
-      tessellator.func_991_a((double)field_9240_E.length, (double)(this.field_6325_d - k), 0.0D);
-      tessellator.func_991_a((double)field_9240_E.length, (double)(this.field_6325_d - k * 2), 0.0D);
+      tessellator.func_991_a(0.0D, this.field_6325_d - k * 2, 0.0D);
+      tessellator.func_991_a(0.0D, this.field_6325_d - k, 0.0D);
+      tessellator.func_991_a(field_9240_E.length, this.field_6325_d - k, 0.0D);
+      tessellator.func_991_a(field_9240_E.length, this.field_6325_d - k * 2, 0.0D);
       tessellator.func_982_a();
       long l = 0L;
 
-      for(int i1 = 0; i1 < field_9240_E.length; ++i1) {
-         l += field_9240_E[i1];
+      for (long value : field_9240_E) {
+         l += value;
       }
 
       int l2 = (int)(l / 200000L / (long)field_9240_E.length);
       tessellator.func_992_a(7);
       tessellator.func_990_b(541065216);
-      tessellator.func_991_a(0.0D, (double)(this.field_6325_d - l2), 0.0D);
-      tessellator.func_991_a(0.0D, (double)this.field_6325_d, 0.0D);
-      tessellator.func_991_a((double)field_9240_E.length, (double)this.field_6325_d, 0.0D);
-      tessellator.func_991_a((double)field_9240_E.length, (double)(this.field_6325_d - l2), 0.0D);
+      tessellator.func_991_a(0.0D, this.field_6325_d - l2, 0.0D);
+      tessellator.func_991_a(0.0D, this.field_6325_d, 0.0D);
+      tessellator.func_991_a(field_9240_E.length, this.field_6325_d, 0.0D);
+      tessellator.func_991_a(field_9240_E.length, this.field_6325_d - l2, 0.0D);
       tessellator.func_982_a();
       tessellator.func_992_a(1);
 
@@ -636,11 +636,11 @@
 
          long j2 = field_9240_E[j1] / 200000L;
          long k2 = field_9239_F[j1] / 200000L;
-         tessellator.func_991_a((double)((float)j1 + 0.5F), (double)((float)((long)this.field_6325_d - j2) + 0.5F), 0.0D);
-         tessellator.func_991_a((double)((float)j1 + 0.5F), (double)((float)this.field_6325_d + 0.5F), 0.0D);
+         tessellator.func_991_a((float)j1 + 0.5F, (float)((long)this.field_6325_d - j2) + 0.5F, 0.0D);
+         tessellator.func_991_a((float)j1 + 0.5F, (float)this.field_6325_d + 0.5F, 0.0D);
          tessellator.func_990_b(-16777216 + l1 * 65536 + l1 * 256 + l1 * 1);
-         tessellator.func_991_a((double)((float)j1 + 0.5F), (double)((float)((long)this.field_6325_d - j2) + 0.5F), 0.0D);
-         tessellator.func_991_a((double)((float)j1 + 0.5F), (double)((float)((long)this.field_6325_d - (j2 - k2)) + 0.5F), 0.0D);
+         tessellator.func_991_a((float)j1 + 0.5F, (float)((long)this.field_6325_d - j2) + 0.5F, 0.0D);
+         tessellator.func_991_a((float)j1 + 0.5F, (float)((long)this.field_6325_d - (j2 - k2)) + 0.5F, 0.0D);
       }
 
       tessellator.func_982_a();
@@ -1148,9 +1148,31 @@
       this.field_6287_N = 0L;
    }
 
-   private void func_6255_d(String param1) {
-      // $FF: Couldn't be decompiled
+   private void func_6255_d(String p_6255_1_) {
+      this.field_6312_q.func_596_a(p_6255_1_);
+      this.field_6312_q.func_595_d("Building terrain");
+      int c1 = '\u0080';
+      byte b = 0;
+      int i = c1 * 2 / 16 + 1;
+      i *= i;
+      for (int c2 = -c1; c2 <= c1; c2 += '\020') {
+         int j = this.field_6324_e.field_1036_o;
+         int k = this.field_6324_e.field_1034_q;
+         if (this.field_6322_g != null) {
+            j = (int)this.field_6322_g.field_611_ak;
+            k = (int)this.field_6322_g.field_609_am;
    }
+         for (int c = -c1; c <= c1; c += '\020') {
+            this.field_6312_q.func_593_a(b++ * 100 / i);
+            this.field_6324_e.func_600_a(j + c2, 64, k + c);
+            while (true) {
+               if (!this.field_6324_e.func_6465_g()) break;
+            }
+         }
+      }
+      this.field_6312_q.func_595_d("Simulating world for a bit");
+      this.field_6324_e.func_656_j();
+   }
 
    public void func_6268_a(String p_6268_1_, File p_6268_2_) {
       int i = p_6268_1_.indexOf("/");
