--- a/net/minecraft/src/BlockRedstoneOre.java
+++ b/net/minecraft/src/BlockRedstoneOre.java
@@ -68,9 +_,9 @@
       double d0 = 0.0625D;
 
       for(int i = 0; i < 6; ++i) {
-         double d1 = (double)((float)p_319_2_ + random.nextFloat());
-         double d2 = (double)((float)p_319_3_ + random.nextFloat());
-         double d3 = (double)((float)p_319_4_ + random.nextFloat());
+         double d1 = (float)p_319_2_ + random.nextFloat();
+         double d2 = (float)p_319_3_ + random.nextFloat();
+         double d3 = (float)p_319_4_ + random.nextFloat();
          if (i == 0 && !p_319_1_.func_601_g(p_319_2_, p_319_3_ + 1, p_319_4_)) {
             d2 = (double)(p_319_3_ + 1) + d0;
          }
