--- a/net/minecraft/src/BlockChest.java
+++ b/net/minecraft/src/BlockChest.java
@@ -165,11 +165,11 @@
                }
 
                itemstack.field_1615_a -= j;
-               EntityItem entityitem = new EntityItem(p_214_1_, (double)((float)p_214_2_ + f), (double)((float)p_214_3_ + f1), (double)((float)p_214_4_ + f2), new ItemStack(itemstack.field_1617_c, j, itemstack.field_1616_d));
+               EntityItem entityitem = new EntityItem(p_214_1_, (float)p_214_2_ + f, (float)p_214_3_ + f1, (float)p_214_4_ + f2, new ItemStack(itemstack.field_1617_c, j, itemstack.field_1616_d));
                float f3 = 0.05F;
-               entityitem.field_608_an = (double)((float)this.field_457_a.nextGaussian() * f3);
-               entityitem.field_607_ao = (double)((float)this.field_457_a.nextGaussian() * f3 + 0.2F);
-               entityitem.field_606_ap = (double)((float)this.field_457_a.nextGaussian() * f3);
+               entityitem.field_608_an = (float)this.field_457_a.nextGaussian() * f3;
+               entityitem.field_607_ao = (float)this.field_457_a.nextGaussian() * f3 + 0.2F;
+               entityitem.field_606_ap = (float)this.field_457_a.nextGaussian() * f3;
                p_214_1_.func_674_a(entityitem);
             }
          }
@@ -179,7 +179,7 @@
    }
 
    public boolean func_250_a(World p_250_1_, int p_250_2_, int p_250_3_, int p_250_4_, EntityPlayer p_250_5_) {
-      Object object = (TileEntityChest)p_250_1_.func_603_b(p_250_2_, p_250_3_, p_250_4_);
+      IInventory iInventory = (TileEntityChest)p_250_1_.func_603_b(p_250_2_, p_250_3_, p_250_4_);
       if (p_250_1_.func_601_g(p_250_2_, p_250_3_ + 1, p_250_4_)) {
          return true;
       } else if (p_250_1_.func_600_a(p_250_2_ - 1, p_250_3_, p_250_4_) == this.field_376_bc && p_250_1_.func_601_g(p_250_2_ - 1, p_250_3_ + 1, p_250_4_)) {
@@ -192,22 +192,22 @@
          return true;
       } else {
          if (p_250_1_.func_600_a(p_250_2_ - 1, p_250_3_, p_250_4_) == this.field_376_bc) {
-            object = new InventoryLargeChest("Large chest", (TileEntityChest)p_250_1_.func_603_b(p_250_2_ - 1, p_250_3_, p_250_4_), (IInventory)object);
+            iInventory = new InventoryLargeChest("Large chest", (TileEntityChest)p_250_1_.func_603_b(p_250_2_ - 1, p_250_3_, p_250_4_), iInventory);
          }
 
          if (p_250_1_.func_600_a(p_250_2_ + 1, p_250_3_, p_250_4_) == this.field_376_bc) {
-            object = new InventoryLargeChest("Large chest", (IInventory)object, (TileEntityChest)p_250_1_.func_603_b(p_250_2_ + 1, p_250_3_, p_250_4_));
+            iInventory = new InventoryLargeChest("Large chest", iInventory, (TileEntityChest)p_250_1_.func_603_b(p_250_2_ + 1, p_250_3_, p_250_4_));
          }
 
          if (p_250_1_.func_600_a(p_250_2_, p_250_3_, p_250_4_ - 1) == this.field_376_bc) {
-            object = new InventoryLargeChest("Large chest", (TileEntityChest)p_250_1_.func_603_b(p_250_2_, p_250_3_, p_250_4_ - 1), (IInventory)object);
+            iInventory = new InventoryLargeChest("Large chest", (TileEntityChest)p_250_1_.func_603_b(p_250_2_, p_250_3_, p_250_4_ - 1), iInventory);
          }
 
          if (p_250_1_.func_600_a(p_250_2_, p_250_3_, p_250_4_ + 1) == this.field_376_bc) {
-            object = new InventoryLargeChest("Large chest", (IInventory)object, (TileEntityChest)p_250_1_.func_603_b(p_250_2_, p_250_3_, p_250_4_ + 1));
+            iInventory = new InventoryLargeChest("Large chest", iInventory, (TileEntityChest)p_250_1_.func_603_b(p_250_2_, p_250_3_, p_250_4_ + 1));
          }
 
-         p_250_5_.func_452_a((IInventory)object);
+         p_250_5_.func_452_a(iInventory);
          return true;
       }
    }
