--- a/net/minecraft/src/BlockFire.java
+++ b/net/minecraft/src/BlockFire.java
@@ -149,7 +149,7 @@
 
    public int func_265_g(World p_265_1_, int p_265_2_, int p_265_3_, int p_265_4_, int p_265_5_) {
       int i = this.field_449_a[p_265_1_.func_600_a(p_265_2_, p_265_3_, p_265_4_)];
-      return i > p_265_5_ ? i : p_265_5_;
+      return Math.max(i, p_265_5_);
    }
 
    public boolean func_243_a(World p_243_1_, int p_243_2_, int p_243_3_, int p_243_4_) {
@@ -174,7 +174,7 @@
 
    public void func_247_b(World p_247_1_, int p_247_2_, int p_247_3_, int p_247_4_, Random p_247_5_) {
       if (p_247_5_.nextInt(24) == 0) {
-         p_247_1_.func_684_a((double)((float)p_247_2_ + 0.5F), (double)((float)p_247_3_ + 0.5F), (double)((float)p_247_4_ + 0.5F), "fire.fire", 1.0F + p_247_5_.nextFloat(), p_247_5_.nextFloat() * 0.7F + 0.3F);
+         p_247_1_.func_684_a((float)p_247_2_ + 0.5F, (float)p_247_3_ + 0.5F, (float)p_247_4_ + 0.5F, "fire.fire", 1.0F + p_247_5_.nextFloat(), p_247_5_.nextFloat() * 0.7F + 0.3F);
       }
 
       if (!p_247_1_.func_601_g(p_247_2_, p_247_3_ - 1, p_247_4_) && !Block.field_402_as.func_261_b(p_247_1_, p_247_2_, p_247_3_ - 1, p_247_4_)) {
@@ -183,7 +183,7 @@
                float f3 = (float)p_247_2_ + p_247_5_.nextFloat() * 0.1F;
                float f8 = (float)p_247_3_ + p_247_5_.nextFloat();
                float f13 = (float)p_247_4_ + p_247_5_.nextFloat();
-               p_247_1_.func_694_a("largesmoke", (double)f3, (double)f8, (double)f13, 0.0D, 0.0D, 0.0D);
+               p_247_1_.func_694_a("largesmoke", f3, f8, f13, 0.0D, 0.0D, 0.0D);
             }
          }
 
@@ -192,7 +192,7 @@
                float f4 = (float)(p_247_2_ + 1) - p_247_5_.nextFloat() * 0.1F;
                float f9 = (float)p_247_3_ + p_247_5_.nextFloat();
                float f14 = (float)p_247_4_ + p_247_5_.nextFloat();
-               p_247_1_.func_694_a("largesmoke", (double)f4, (double)f9, (double)f14, 0.0D, 0.0D, 0.0D);
+               p_247_1_.func_694_a("largesmoke", f4, f9, f14, 0.0D, 0.0D, 0.0D);
             }
          }
 
@@ -201,7 +201,7 @@
                float f5 = (float)p_247_2_ + p_247_5_.nextFloat();
                float f10 = (float)p_247_3_ + p_247_5_.nextFloat();
                float f15 = (float)p_247_4_ + p_247_5_.nextFloat() * 0.1F;
-               p_247_1_.func_694_a("largesmoke", (double)f5, (double)f10, (double)f15, 0.0D, 0.0D, 0.0D);
+               p_247_1_.func_694_a("largesmoke", f5, f10, f15, 0.0D, 0.0D, 0.0D);
             }
          }
 
@@ -210,7 +210,7 @@
                float f6 = (float)p_247_2_ + p_247_5_.nextFloat();
                float f11 = (float)p_247_3_ + p_247_5_.nextFloat();
                float f16 = (float)(p_247_4_ + 1) - p_247_5_.nextFloat() * 0.1F;
-               p_247_1_.func_694_a("largesmoke", (double)f6, (double)f11, (double)f16, 0.0D, 0.0D, 0.0D);
+               p_247_1_.func_694_a("largesmoke", f6, f11, f16, 0.0D, 0.0D, 0.0D);
             }
          }
 
@@ -219,7 +219,7 @@
                float f7 = (float)p_247_2_ + p_247_5_.nextFloat();
                float f12 = (float)(p_247_3_ + 1) - p_247_5_.nextFloat() * 0.1F;
                float f17 = (float)p_247_4_ + p_247_5_.nextFloat();
-               p_247_1_.func_694_a("largesmoke", (double)f7, (double)f12, (double)f17, 0.0D, 0.0D, 0.0D);
+               p_247_1_.func_694_a("largesmoke", f7, f12, f17, 0.0D, 0.0D, 0.0D);
             }
          }
       } else {
@@ -227,7 +227,7 @@
             float f = (float)p_247_2_ + p_247_5_.nextFloat();
             float f1 = (float)p_247_3_ + p_247_5_.nextFloat() * 0.5F + 0.5F;
             float f2 = (float)p_247_4_ + p_247_5_.nextFloat();
-            p_247_1_.func_694_a("largesmoke", (double)f, (double)f1, (double)f2, 0.0D, 0.0D, 0.0D);
+            p_247_1_.func_694_a("largesmoke", f, f1, f2, 0.0D, 0.0D, 0.0D);
          }
       }
 
