--- a/net/minecraft/src/BlockFurnace.java
+++ b/net/minecraft/src/BlockFurnace.java
@@ -69,17 +69,17 @@
          float f3 = 0.52F;
          float f4 = p_247_5_.nextFloat() * 0.6F - 0.3F;
          if (i == 4) {
-            p_247_1_.func_694_a("smoke", (double)(f - f3), (double)f1, (double)(f2 + f4), 0.0D, 0.0D, 0.0D);
-            p_247_1_.func_694_a("flame", (double)(f - f3), (double)f1, (double)(f2 + f4), 0.0D, 0.0D, 0.0D);
+            p_247_1_.func_694_a("smoke", f - f3, f1, f2 + f4, 0.0D, 0.0D, 0.0D);
+            p_247_1_.func_694_a("flame", f - f3, f1, f2 + f4, 0.0D, 0.0D, 0.0D);
          } else if (i == 5) {
-            p_247_1_.func_694_a("smoke", (double)(f + f3), (double)f1, (double)(f2 + f4), 0.0D, 0.0D, 0.0D);
-            p_247_1_.func_694_a("flame", (double)(f + f3), (double)f1, (double)(f2 + f4), 0.0D, 0.0D, 0.0D);
+            p_247_1_.func_694_a("smoke", f + f3, f1, f2 + f4, 0.0D, 0.0D, 0.0D);
+            p_247_1_.func_694_a("flame", f + f3, f1, f2 + f4, 0.0D, 0.0D, 0.0D);
          } else if (i == 2) {
-            p_247_1_.func_694_a("smoke", (double)(f + f4), (double)f1, (double)(f2 - f3), 0.0D, 0.0D, 0.0D);
-            p_247_1_.func_694_a("flame", (double)(f + f4), (double)f1, (double)(f2 - f3), 0.0D, 0.0D, 0.0D);
+            p_247_1_.func_694_a("smoke", f + f4, f1, f2 - f3, 0.0D, 0.0D, 0.0D);
+            p_247_1_.func_694_a("flame", f + f4, f1, f2 - f3, 0.0D, 0.0D, 0.0D);
          } else if (i == 3) {
-            p_247_1_.func_694_a("smoke", (double)(f + f4), (double)f1, (double)(f2 + f3), 0.0D, 0.0D, 0.0D);
-            p_247_1_.func_694_a("flame", (double)(f + f4), (double)f1, (double)(f2 + f3), 0.0D, 0.0D, 0.0D);
+            p_247_1_.func_694_a("smoke", f + f4, f1, f2 + f3, 0.0D, 0.0D, 0.0D);
+            p_247_1_.func_694_a("flame", f + f4, f1, f2 + f3, 0.0D, 0.0D, 0.0D);
          }
 
       }
@@ -119,7 +119,7 @@
    }
 
    public void func_4026_a(World p_4026_1_, int p_4026_2_, int p_4026_3_, int p_4026_4_, EntityLiving p_4026_5_) {
-      int i = MathHelper.func_1108_b((double)(p_4026_5_.field_605_aq * 4.0F / 360.0F) + 0.5D) & 3;
+      int i = MathHelper.func_1108_b(p_4026_5_.field_605_aq * 4.0F / 360.0F + 0.5D) & 3;
       if (i == 0) {
          p_4026_1_.func_691_b(p_4026_2_, p_4026_3_, p_4026_4_, 2);
       }
