--- a/net/minecraft/src/ChunkProviderClient.java
+++ b/net/minecraft/src/ChunkProviderClient.java
@@ -8,8 +8,8 @@
 
 public class ChunkProviderClient implements IChunkProvider {
    private Chunk field_887_a;
-   private Map field_886_b = new HashMap();
-   private List field_889_c = new ArrayList();
+   private Map<ChunkCoordinates, Chunk> field_886_b = new HashMap<>();
+   private List field_889_c = new ArrayList<>();
    private World field_888_d;
 
    public ChunkProviderClient(World p_i270_1_) {
@@ -46,7 +46,7 @@
 
    public Chunk func_533_b(int p_533_1_, int p_533_2_) {
       ChunkCoordinates chunkcoordinates = new ChunkCoordinates(p_533_1_, p_533_2_);
-      Chunk chunk = (Chunk)this.field_886_b.get(chunkcoordinates);
+      Chunk chunk = this.field_886_b.get(chunkcoordinates);
       return chunk == null ? this.field_887_a : chunk;
    }
 
