--- a/net/minecraft/src/EntityLiving.java
+++ b/net/minecraft/src/EntityLiving.java
@@ -189,8 +_,8 @@
       this.field_9323_Z = p_378_1_;
       this.field_9356_aa = p_378_3_;
       this.field_9354_ab = p_378_5_;
-      this.field_9352_ac = (double)p_378_7_;
-      this.field_9350_ad = (double)p_378_8_;
+      this.field_9352_ac = p_378_7_;
+      this.field_9350_ad = p_378_8_;
       this.field_9324_Y = p_378_9_;
    }
 
@@ -390,10 +_,10 @@
       this.field_607_ao /= 2.0D;
       this.field_606_ap /= 2.0D;
       this.field_608_an -= p_434_3_ / (double)f * (double)f1;
-      this.field_607_ao += (double)0.4F;
+      this.field_607_ao += 0.4F;
       this.field_606_ap -= p_434_5_ / (double)f * (double)f1;
       if (this.field_607_ao > (double)0.4F) {
-         this.field_607_ao = (double)0.4F;
+         this.field_607_ao = 0.4F;
       }
 
    }
@@ -423,7 +_,7 @@
    }
 
    protected void func_400_c(float p_400_1_) {
-      int i = (int)Math.ceil((double)(p_400_1_ - 3.0F));
+      int i = (int)Math.ceil(p_400_1_ - 3.0F);
       if (i > 0) {
          this.func_396_a((Entity)null, i);
          int j = this.field_615_ag.func_600_a(MathHelper.func_1108_b(this.field_611_ak), MathHelper.func_1108_b(this.field_610_al - (double)0.2F - (double)this.field_9292_aO), MathHelper.func_1108_b(this.field_609_am));
@@ -440,12 +_,12 @@
          double d0 = this.field_610_al;
          this.func_351_a(p_435_1_, p_435_2_, 0.02F);
          this.func_349_c(this.field_608_an, this.field_607_ao, this.field_606_ap);
-         this.field_608_an *= (double)0.8F;
-         this.field_607_ao *= (double)0.8F;
-         this.field_606_ap *= (double)0.8F;
+         this.field_608_an *= 0.8F;
+         this.field_607_ao *= 0.8F;
+         this.field_606_ap *= 0.8F;
          this.field_607_ao -= 0.02D;
          if (this.field_9297_aI && this.func_403_b(this.field_608_an, this.field_607_ao + (double)0.6F - this.field_610_al + d0, this.field_606_ap)) {
-            this.field_607_ao = (double)0.3F;
+            this.field_607_ao = 0.3F;
          }
       } else if (this.func_359_G()) {
          double d1 = this.field_610_al;
@@ -456,7 +_,7 @@
          this.field_606_ap *= 0.5D;
          this.field_607_ao -= 0.02D;
          if (this.field_9297_aI && this.func_403_b(this.field_608_an, this.field_607_ao + (double)0.6F - this.field_610_al + d1, this.field_606_ap)) {
-            this.field_607_ao = (double)0.3F;
+            this.field_607_ao = 0.3F;
          }
       } else {
          float f1 = 0.91F;
@@ -492,9 +_,9 @@
          }
 
          this.field_607_ao -= 0.08D;
-         this.field_607_ao *= (double)0.98F;
-         this.field_608_an *= (double)f1;
-         this.field_606_ap *= (double)f1;
+         this.field_607_ao *= 0.98F;
+         this.field_608_an *= f1;
+         this.field_606_ap *= f1;
       }
 
       this.field_705_Q = this.field_704_R;
@@ -572,9 +_,9 @@
       boolean flag = this.func_359_G();
       if (this.field_9336_ak) {
          if (flag1) {
-            this.field_607_ao += (double)0.04F;
+            this.field_607_ao += 0.04F;
          } else if (flag) {
-            this.field_607_ao += (double)0.04F;
+            this.field_607_ao += 0.04F;
          } else if (this.field_9298_aH) {
             this.func_424_C();
          }
@@ -584,10 +_,9 @@
       this.field_9340_ai *= 0.98F;
       this.field_9338_aj *= 0.9F;
       this.func_435_b(this.field_9342_ah, this.field_9340_ai);
-      List list = this.field_615_ag.func_659_b(this, this.field_601_au.func_1177_b((double)0.2F, 0.0D, (double)0.2F));
+      List<Entity> list = this.field_615_ag.func_659_b(this, this.field_601_au.func_1177_b(0.2F, 0.0D, 0.2F));
       if (list != null && list.size() > 0) {
-         for(int i = 0; i < list.size(); ++i) {
-            Entity entity = (Entity)list.get(i);
+         for (Entity entity : list) {
             if (entity.func_385_d_()) {
                entity.func_380_f(this);
             }
@@ -597,7 +_,7 @@
    }
 
    protected void func_424_C() {
-      this.field_607_ao = (double)0.42F;
+      this.field_607_ao = 0.42F;
    }
 
    protected void func_418_b_() {
@@ -625,7 +_,7 @@
       this.field_9340_ai = 0.0F;
       float f = 8.0F;
       if (this.field_9312_bd.nextFloat() < 0.02F) {
-         entityplayer = this.field_615_ag.func_609_a(this, (double)f);
+         entityplayer = this.field_615_ag.func_609_a(this, f);
          if (entityplayer != null) {
             this.field_4120_b = entityplayer;
             this.field_4127_c = 10 + this.field_9312_bd.nextInt(20);
@@ -667,7 +_,7 @@
          d1 = (p_426_1_.field_601_au.field_1697_b + p_426_1_.field_601_au.field_1702_e) / 2.0D - (this.field_610_al + (double)this.func_373_s());
       }
 
-      double d3 = (double)MathHelper.func_1109_a(d0 * d0 + d2 * d2);
+      double d3 = MathHelper.func_1109_a(d0 * d0 + d2 * d2);
       float f = (float)(Math.atan2(d2, d0) * 180.0D / (double)(float)Math.PI) - 90.0F;
       float f1 = (float)(Math.atan2(d1, d3) * 180.0D / (double)(float)Math.PI);
       this.field_604_ar = -this.func_417_b(this.field_604_ar, f1, p_426_2_);
@@ -735,7 +_,7 @@
          float f7 = MathHelper.func_1106_a(-this.field_605_aq * ((float)Math.PI / 180F) - (float)Math.PI);
          float f8 = -MathHelper.func_1114_b(-this.field_604_ar * ((float)Math.PI / 180F));
          float f9 = MathHelper.func_1106_a(-this.field_604_ar * ((float)Math.PI / 180F));
-         return Vec3D.func_1248_b((double)(f7 * f8), (double)f9, (double)(f6 * f8));
+         return Vec3D.func_1248_b(f7 * f8, f9, f6 * f8);
       } else {
          float f = this.field_602_at + (this.field_604_ar - this.field_602_at) * p_430_1_;
          float f1 = this.field_603_as + (this.field_605_aq - this.field_603_as) * p_430_1_;
@@ -743,7 +_,7 @@
          float f3 = MathHelper.func_1106_a(-f1 * ((float)Math.PI / 180F) - (float)Math.PI);
          float f4 = -MathHelper.func_1114_b(-f * ((float)Math.PI / 180F));
          float f5 = MathHelper.func_1106_a(-f * ((float)Math.PI / 180F));
-         return Vec3D.func_1248_b((double)(f3 * f4), (double)f5, (double)(f2 * f4));
+         return Vec3D.func_1248_b(f3 * f4, f5, f2 * f4);
       }
    }
 
