--- a/net/minecraft/src/EntityArrow.java
+++ b/net/minecraft/src/EntityArrow.java
@@ -30,14 +30,14 @@
       this.field_682_g = p_i239_2_;
       this.func_371_a(0.5F, 0.5F);
       this.func_365_c(p_i239_2_.field_611_ak, p_i239_2_.field_610_al + (double)p_i239_2_.func_373_s(), p_i239_2_.field_609_am, p_i239_2_.field_605_aq, p_i239_2_.field_604_ar);
-      this.field_611_ak -= (double)(MathHelper.func_1114_b(this.field_605_aq / 180.0F * (float)Math.PI) * 0.16F);
-      this.field_610_al -= (double)0.1F;
-      this.field_609_am -= (double)(MathHelper.func_1106_a(this.field_605_aq / 180.0F * (float)Math.PI) * 0.16F);
+      this.field_611_ak -= MathHelper.func_1114_b(this.field_605_aq / 180.0F * (float)Math.PI) * 0.16F;
+      this.field_610_al -= 0.1F;
+      this.field_609_am -= MathHelper.func_1106_a(this.field_605_aq / 180.0F * (float)Math.PI) * 0.16F;
       this.func_347_a(this.field_611_ak, this.field_610_al, this.field_609_am);
       this.field_9292_aO = 0.0F;
-      this.field_608_an = (double)(-MathHelper.func_1106_a(this.field_605_aq / 180.0F * (float)Math.PI) * MathHelper.func_1114_b(this.field_604_ar / 180.0F * (float)Math.PI));
-      this.field_606_ap = (double)(MathHelper.func_1114_b(this.field_605_aq / 180.0F * (float)Math.PI) * MathHelper.func_1114_b(this.field_604_ar / 180.0F * (float)Math.PI));
-      this.field_607_ao = (double)(-MathHelper.func_1106_a(this.field_604_ar / 180.0F * (float)Math.PI));
+      this.field_608_an = -MathHelper.func_1106_a(this.field_605_aq / 180.0F * (float)Math.PI) * MathHelper.func_1114_b(this.field_604_ar / 180.0F * (float)Math.PI);
+      this.field_606_ap = MathHelper.func_1114_b(this.field_605_aq / 180.0F * (float)Math.PI) * MathHelper.func_1114_b(this.field_604_ar / 180.0F * (float)Math.PI);
+      this.field_607_ao = -MathHelper.func_1106_a(this.field_604_ar / 180.0F * (float)Math.PI);
       this.func_408_a(this.field_608_an, this.field_607_ao, this.field_606_ap, 1.5F, 1.0F);
    }
 
@@ -57,7 +57,7 @@
       this.field_606_ap = p_408_5_;
       float f1 = MathHelper.func_1109_a(p_408_1_ * p_408_1_ + p_408_5_ * p_408_5_);
       this.field_603_as = this.field_605_aq = (float)(Math.atan2(p_408_1_, p_408_5_) * 180.0D / (double)(float)Math.PI);
-      this.field_602_at = this.field_604_ar = (float)(Math.atan2(p_408_3_, (double)f1) * 180.0D / (double)(float)Math.PI);
+      this.field_602_at = this.field_604_ar = (float)(Math.atan2(p_408_3_, f1) * 180.0D / (double)(float)Math.PI);
       this.field_681_h = 0;
    }
 
@@ -68,7 +68,7 @@
       if (this.field_602_at == 0.0F && this.field_603_as == 0.0F) {
          float f = MathHelper.func_1109_a(p_6375_1_ * p_6375_1_ + p_6375_5_ * p_6375_5_);
          this.field_603_as = this.field_605_aq = (float)(Math.atan2(p_6375_1_, p_6375_5_) * 180.0D / (double)(float)Math.PI);
-         this.field_602_at = this.field_604_ar = (float)(Math.atan2(p_6375_3_, (double)f) * 180.0D / (double)(float)Math.PI);
+         this.field_602_at = this.field_604_ar = (float)(Math.atan2(p_6375_3_, f) * 180.0D / (double)(float)Math.PI);
       }
 
    }
@@ -78,7 +78,7 @@
       if (this.field_602_at == 0.0F && this.field_603_as == 0.0F) {
          float f = MathHelper.func_1109_a(this.field_608_an * this.field_608_an + this.field_606_ap * this.field_606_ap);
          this.field_603_as = this.field_605_aq = (float)(Math.atan2(this.field_608_an, this.field_606_ap) * 180.0D / (double)(float)Math.PI);
-         this.field_602_at = this.field_604_ar = (float)(Math.atan2(this.field_607_ao, (double)f) * 180.0D / (double)(float)Math.PI);
+         this.field_602_at = this.field_604_ar = (float)(Math.atan2(this.field_607_ao, f) * 180.0D / (double)(float)Math.PI);
       }
 
       if (this.field_688_a > 0) {
@@ -97,9 +97,9 @@
          }
 
          this.field_9315_g = false;
-         this.field_608_an *= (double)(this.field_9312_bd.nextFloat() * 0.2F);
-         this.field_607_ao *= (double)(this.field_9312_bd.nextFloat() * 0.2F);
-         this.field_606_ap *= (double)(this.field_9312_bd.nextFloat() * 0.2F);
+         this.field_608_an *= (this.field_9312_bd.nextFloat() * 0.2F);
+         this.field_607_ao *= (this.field_9312_bd.nextFloat() * 0.2F);
+         this.field_606_ap *= (this.field_9312_bd.nextFloat() * 0.2F);
          this.field_681_h = 0;
          this.field_680_i = 0;
       } else {
@@ -116,14 +116,13 @@
       }
 
       Entity entity = null;
-      List list = this.field_615_ag.func_659_b(this, this.field_601_au.func_1170_a(this.field_608_an, this.field_607_ao, this.field_606_ap).func_1177_b(1.0D, 1.0D, 1.0D));
+      List<Entity> list = this.field_615_ag.func_659_b(this, this.field_601_au.func_1170_a(this.field_608_an, this.field_607_ao, this.field_606_ap).func_1177_b(1.0D, 1.0D, 1.0D));
       double d0 = 0.0D;
 
-      for(int i = 0; i < list.size(); ++i) {
-         Entity entity1 = (Entity)list.get(i);
+      for (Entity entity1 : list) {
          if (entity1.func_401_c_() && (entity1 != this.field_682_g || this.field_680_i >= 5)) {
             float f1 = 0.3F;
-            AxisAlignedBB axisalignedbb = entity1.field_601_au.func_1177_b((double)f1, (double)f1, (double)f1);
+            AxisAlignedBB axisalignedbb = entity1.field_601_au.func_1177_b(f1, f1, f1);
             MovingObjectPosition movingobjectposition1 = axisalignedbb.func_1169_a(vec3d1, vec3d);
             if (movingobjectposition1 != null) {
                double d1 = vec3d1.func_1251_c(movingobjectposition1.field_1169_f);
@@ -145,9 +144,9 @@
                this.field_615_ag.func_623_a(this, "random.drr", 1.0F, 1.2F / (this.field_9312_bd.nextFloat() * 0.2F + 0.9F));
                this.func_395_F();
             } else {
-               this.field_608_an *= (double)-0.1F;
-               this.field_607_ao *= (double)-0.1F;
-               this.field_606_ap *= (double)-0.1F;
+               this.field_608_an *= -0.1F;
+               this.field_607_ao *= -0.1F;
+               this.field_606_ap *= -0.1F;
                this.field_605_aq += 180.0F;
                this.field_603_as += 180.0F;
                this.field_680_i = 0;
@@ -157,9 +156,9 @@
             this.field_9318_d = movingobjectposition.field_1172_c;
             this.field_9317_e = movingobjectposition.field_1171_d;
             this.field_9316_f = this.field_615_ag.func_600_a(this.field_9319_c, this.field_9318_d, this.field_9317_e);
-            this.field_608_an = (double)((float)(movingobjectposition.field_1169_f.field_1776_a - this.field_611_ak));
-            this.field_607_ao = (double)((float)(movingobjectposition.field_1169_f.field_1775_b - this.field_610_al));
-            this.field_606_ap = (double)((float)(movingobjectposition.field_1169_f.field_1779_c - this.field_609_am));
+            this.field_608_an = (float)(movingobjectposition.field_1169_f.field_1776_a - this.field_611_ak);
+            this.field_607_ao = (float)(movingobjectposition.field_1169_f.field_1775_b - this.field_610_al);
+            this.field_606_ap = (float)(movingobjectposition.field_1169_f.field_1779_c - this.field_609_am);
             float f2 = MathHelper.func_1109_a(this.field_608_an * this.field_608_an + this.field_607_ao * this.field_607_ao + this.field_606_ap * this.field_606_ap);
             this.field_611_ak -= this.field_608_an / (double)f2 * (double)0.05F;
             this.field_610_al -= this.field_607_ao / (double)f2 * (double)0.05F;
@@ -176,7 +175,7 @@
       float f3 = MathHelper.func_1109_a(this.field_608_an * this.field_608_an + this.field_606_ap * this.field_606_ap);
       this.field_605_aq = (float)(Math.atan2(this.field_608_an, this.field_606_ap) * 180.0D / (double)(float)Math.PI);
 
-      for(this.field_604_ar = (float)(Math.atan2(this.field_607_ao, (double)f3) * 180.0D / (double)(float)Math.PI); this.field_604_ar - this.field_602_at < -180.0F; this.field_602_at -= 360.0F) {
+      for(this.field_604_ar = (float)(Math.atan2(this.field_607_ao, f3) * 180.0D / (double)(float)Math.PI); this.field_604_ar - this.field_602_at < -180.0F; this.field_602_at -= 360.0F) {
       }
 
       while(this.field_604_ar - this.field_602_at >= 180.0F) {
@@ -204,10 +203,10 @@
          f4 = 0.8F;
       }
 
-      this.field_608_an *= (double)f4;
-      this.field_607_ao *= (double)f4;
-      this.field_606_ap *= (double)f4;
-      this.field_607_ao -= (double)f5;
+      this.field_608_an *= f4;
+      this.field_607_ao *= f4;
+      this.field_606_ap *= f4;
+      this.field_607_ao -= f5;
       this.func_347_a(this.field_611_ak, this.field_610_al, this.field_609_am);
    }
 
