--- a/net/minecraft/src/Gui.java
+++ b/net/minecraft/src/Gui.java
@@ -16,10 +16,10 @@
       GL11.glBlendFunc(770, 771);
       GL11.glColor4f(f1, f2, f3, f);
       tessellator.func_977_b();
-      tessellator.func_991_a((double)p_551_1_, (double)p_551_4_, 0.0D);
-      tessellator.func_991_a((double)p_551_3_, (double)p_551_4_, 0.0D);
-      tessellator.func_991_a((double)p_551_3_, (double)p_551_2_, 0.0D);
-      tessellator.func_991_a((double)p_551_1_, (double)p_551_2_, 0.0D);
+      tessellator.func_991_a(p_551_1_, p_551_4_, 0.0D);
+      tessellator.func_991_a(p_551_3_, p_551_4_, 0.0D);
+      tessellator.func_991_a(p_551_3_, p_551_2_, 0.0D);
+      tessellator.func_991_a(p_551_1_, p_551_2_, 0.0D);
       tessellator.func_982_a();
       GL11.glEnable(3553);
       GL11.glDisable(3042);
@@ -42,11 +42,11 @@
       Tessellator tessellator = Tessellator.field_1512_a;
       tessellator.func_977_b();
       tessellator.func_986_a(f1, f2, f3, f);
-      tessellator.func_991_a((double)p_549_3_, (double)p_549_2_, 0.0D);
-      tessellator.func_991_a((double)p_549_1_, (double)p_549_2_, 0.0D);
+      tessellator.func_991_a(p_549_3_, p_549_2_, 0.0D);
+      tessellator.func_991_a(p_549_1_, p_549_2_, 0.0D);
       tessellator.func_986_a(f5, f6, f7, f4);
-      tessellator.func_991_a((double)p_549_1_, (double)p_549_4_, 0.0D);
-      tessellator.func_991_a((double)p_549_3_, (double)p_549_4_, 0.0D);
+      tessellator.func_991_a(p_549_1_, p_549_4_, 0.0D);
+      tessellator.func_991_a(p_549_3_, p_549_4_, 0.0D);
       tessellator.func_982_a();
       GL11.glShadeModel(7424);
       GL11.glDisable(3042);
@@ -67,10 +67,10 @@
       float f1 = 0.00390625F;
       Tessellator tessellator = Tessellator.field_1512_a;
       tessellator.func_977_b();
-      tessellator.func_983_a((double)(p_550_1_ + 0), (double)(p_550_2_ + p_550_6_), (double)this.field_923_k, (double)((float)(p_550_3_ + 0) * f), (double)((float)(p_550_4_ + p_550_6_) * f1));
-      tessellator.func_983_a((double)(p_550_1_ + p_550_5_), (double)(p_550_2_ + p_550_6_), (double)this.field_923_k, (double)((float)(p_550_3_ + p_550_5_) * f), (double)((float)(p_550_4_ + p_550_6_) * f1));
-      tessellator.func_983_a((double)(p_550_1_ + p_550_5_), (double)(p_550_2_ + 0), (double)this.field_923_k, (double)((float)(p_550_3_ + p_550_5_) * f), (double)((float)(p_550_4_ + 0) * f1));
-      tessellator.func_983_a((double)(p_550_1_ + 0), (double)(p_550_2_ + 0), (double)this.field_923_k, (double)((float)(p_550_3_ + 0) * f), (double)((float)(p_550_4_ + 0) * f1));
+      tessellator.func_983_a(p_550_1_, p_550_2_ + p_550_6_, this.field_923_k, (float)(p_550_3_) * f, (float)(p_550_4_ + p_550_6_) * f1);
+      tessellator.func_983_a(p_550_1_ + p_550_5_, p_550_2_ + p_550_6_, this.field_923_k, (float)(p_550_3_ + p_550_5_) * f, (float)(p_550_4_ + p_550_6_) * f1);
+      tessellator.func_983_a(p_550_1_ + p_550_5_, p_550_2_, this.field_923_k, (float)(p_550_3_ + p_550_5_) * f, (float)(p_550_4_) * f1);
+      tessellator.func_983_a(p_550_1_, p_550_2_, this.field_923_k, (float)(p_550_3_) * f, (float)(p_550_4_) * f1);
       tessellator.func_982_a();
    }
 }
