--- a/net/minecraft/src/GuiEditSign.java
+++ b/net/minecraft/src/GuiEditSign.java
@@ -68,7 +_,6 @@
       if (block == Block.field_443_aE) {
          float f1 = (float)(this.field_1002_h.func_479_f() * 360) / 16.0F;
          GL11.glRotatef(f1, 0.0F, 1.0F, 0.0F);
-         GL11.glTranslatef(0.0F, 0.3125F, 0.0F);
       } else {
          int i = this.field_1002_h.func_479_f();
          float f2 = 0.0F;
@@ -85,8 +_,8 @@
          }
 
          GL11.glRotatef(f2, 0.0F, 1.0F, 0.0F);
-         GL11.glTranslatef(0.0F, 0.3125F, 0.0F);
       }
+      GL11.glTranslatef(0.0F, 0.3125F, 0.0F);
 
       if (this.field_4189_i / 6 % 2 == 0) {
          this.field_1002_h.field_825_b = this.field_1000_j;
