--- a/net/minecraft/src/GuiMainMenu.java
+++ b/net/minecraft/src/GuiMainMenu.java
@@ -18,7 +18,7 @@
 
    public GuiMainMenu() {
       try {
-         ArrayList arraylist = new ArrayList();
+         ArrayList<String> arraylist = new ArrayList<>();
          BufferedReader bufferedreader = new BufferedReader(new InputStreamReader(GuiMainMenu.class.getResourceAsStream("/title/splashes.txt")));
          String s = "";
 
@@ -29,7 +29,7 @@
             }
          }
 
-         this.field_6462_l = (String)arraylist.get(field_6463_h.nextInt(arraylist.size()));
+         this.field_6462_l = arraylist.get(field_6463_h.nextInt(arraylist.size()));
       } catch (Exception exception) {
       }
 
@@ -38,9 +38,9 @@
    public void func_570_g() {
       ++this.field_989_j;
       if (this.field_990_i != null) {
-         for(int i = 0; i < this.field_990_i.length; ++i) {
-            for(int j = 0; j < this.field_990_i[i].length; ++j) {
-               this.field_990_i[i][j].func_875_a();
+         for (LogoEffectRandomizer[] logoEffectRandomizers : this.field_990_i) {
+            for (LogoEffectRandomizer logoEffectRandomizer : logoEffectRandomizers) {
+               logoEffectRandomizer.func_875_a();
             }
          }
       }
@@ -68,7 +68,7 @@
       this.field_949_e.add(new GuiButton(3, this.field_951_c / 2 - 100, this.field_950_d / 4 + 96, "Mods and Texture Packs"));
       this.field_949_e.add(new GuiButton(0, this.field_951_c / 2 - 100, this.field_950_d / 4 + 120 + 12, "Options..."));
       if (this.field_945_b.field_6320_i == null) {
-         ((GuiButton)this.field_949_e.get(1)).field_937_g = false;
+         this.field_949_e.get(1).field_937_g = false;
       }
 
    }
