--- a/net/minecraft/src/ItemBoat.java
+++ b/net/minecraft/src/ItemBoat.java
@@ -23,21 +_,19 @@
       double d3 = 5.0D;
       Vec3D vec3d1 = vec3d.func_1257_c((double)f7 * d3, (double)f6 * d3, (double)f8 * d3);
       MovingObjectPosition movingobjectposition = p_193_2_.func_642_a(vec3d, vec3d1, true);
-      if (movingobjectposition == null) {
-         return p_193_1_;
-      } else {
+      if (movingobjectposition != null) {
          if (movingobjectposition.field_1167_a == 0) {
             int i = movingobjectposition.field_1166_b;
             int j = movingobjectposition.field_1172_c;
             int k = movingobjectposition.field_1171_d;
             if (!p_193_2_.field_1026_y) {
-               p_193_2_.func_674_a(new EntityBoat(p_193_2_, (double)((float)i + 0.5F), (double)((float)j + 1.5F), (double)((float)k + 0.5F)));
+               p_193_2_.func_674_a(new EntityBoat(p_193_2_, (float) i + 0.5F, (float) j + 1.5F, (float) k + 0.5F));
             }
 
             --p_193_1_.field_1615_a;
          }
 
+      }
          return p_193_1_;
       }
    }
-}
