--- a/net/minecraft/src/NetClientHandler.java
+++ b/net/minecraft/src/NetClientHandler.java
@@ -58,47 +_,47 @@
       double d0 = (double)p_835_1_.field_499_b / 32.0D;
       double d1 = (double)p_835_1_.field_503_c / 32.0D;
       double d2 = (double)p_835_1_.field_502_d / 32.0D;
-      Object object = null;
+      Entity entity = null;
       if (p_835_1_.field_501_e == 10) {
-         object = new EntityMinecart(this.field_1211_f, d0, d1, d2, 0);
+         entity = new EntityMinecart(this.field_1211_f, d0, d1, d2, 0);
       }
 
       if (p_835_1_.field_501_e == 11) {
-         object = new EntityMinecart(this.field_1211_f, d0, d1, d2, 1);
+         entity = new EntityMinecart(this.field_1211_f, d0, d1, d2, 1);
       }
 
       if (p_835_1_.field_501_e == 12) {
-         object = new EntityMinecart(this.field_1211_f, d0, d1, d2, 2);
+         entity = new EntityMinecart(this.field_1211_f, d0, d1, d2, 2);
       }
 
       if (p_835_1_.field_501_e == 90) {
-         object = new EntityFish(this.field_1211_f, d0, d1, d2);
+         entity = new EntityFish(this.field_1211_f, d0, d1, d2);
       }
 
       if (p_835_1_.field_501_e == 60) {
-         object = new EntityArrow(this.field_1211_f, d0, d1, d2);
+         entity = new EntityArrow(this.field_1211_f, d0, d1, d2);
       }
 
       if (p_835_1_.field_501_e == 61) {
-         object = new EntitySnowball(this.field_1211_f, d0, d1, d2);
+         entity = new EntitySnowball(this.field_1211_f, d0, d1, d2);
       }
 
       if (p_835_1_.field_501_e == 1) {
-         object = new EntityBoat(this.field_1211_f, d0, d1, d2);
+         entity = new EntityBoat(this.field_1211_f, d0, d1, d2);
       }
 
       if (p_835_1_.field_501_e == 50) {
-         object = new EntityTNTPrimed(this.field_1211_f, d0, d1, d2);
+         entity = new EntityTNTPrimed(this.field_1211_f, d0, d1, d2);
       }
 
-      if (object != null) {
-         ((Entity)object).field_9303_br = p_835_1_.field_499_b;
-         ((Entity)object).field_9302_bs = p_835_1_.field_503_c;
-         ((Entity)object).field_9301_bt = p_835_1_.field_502_d;
-         ((Entity)object).field_605_aq = 0.0F;
-         ((Entity)object).field_604_ar = 0.0F;
-         ((Entity)object).field_620_ab = p_835_1_.field_500_a;
-         this.field_1211_f.func_712_a(p_835_1_.field_500_a, (Entity)object);
+      if (entity != null) {
+         entity.field_9303_br = p_835_1_.field_499_b;
+         entity.field_9302_bs = p_835_1_.field_503_c;
+         entity.field_9301_bt = p_835_1_.field_502_d;
+         entity.field_605_aq = 0.0F;
+         entity.field_604_ar = 0.0F;
+         entity.field_620_ab = p_835_1_.field_500_a;
+         this.field_1211_f.func_712_a(p_835_1_.field_500_a, entity);
       }
 
    }
@@ -256,14 +_,14 @@
 
    public void func_834_a(Packet22Collect p_834_1_) {
       Entity entity = this.func_12246_a(p_834_1_.field_581_a);
-      Object object = (EntityLiving)this.func_12246_a(p_834_1_.field_580_b);
-      if (object == null) {
-         object = this.field_1212_e.field_6322_g;
+      EntityLiving entityLiving = (EntityLiving)this.func_12246_a(p_834_1_.field_580_b);
+      if (entityLiving == null) {
+         entityLiving = this.field_1212_e.field_6322_g;
       }
 
       if (entity != null) {
          this.field_1211_f.func_623_a(entity, "random.pop", 0.2F, ((this.field_1208_b.nextFloat() - this.field_1208_b.nextFloat()) * 0.7F + 1.0F) * 2.0F);
-         this.field_1212_e.field_6321_h.func_1192_a(new EntityPickupFX(this.field_1212_e.field_6324_e, entity, (Entity)object, -0.5F));
+         this.field_1212_e.field_6321_h.func_1192_a(new EntityPickupFX(this.field_1212_e.field_6324_e, entity, entityLiving, -0.5F));
          this.field_1211_f.func_710_c(p_834_1_.field_581_a);
       }
 
@@ -332,7 +_,7 @@
             }
          } catch (Exception exception) {
             exception.printStackTrace();
-            this.field_1213_d.func_974_a("Internal client error: " + exception.toString());
+            this.field_1213_d.func_974_a("Internal client error: " + exception);
          }
       }
 
@@ -405,14 +_,14 @@
    }
 
    public void func_6497_a(Packet39 p_6497_1_) {
-      Object object = this.func_12246_a(p_6497_1_.field_6365_a);
-      Entity entity = this.func_12246_a(p_6497_1_.field_6364_b);
+      Entity entity = this.func_12246_a(p_6497_1_.field_6365_a);
+      Entity entity1 = this.func_12246_a(p_6497_1_.field_6364_b);
       if (p_6497_1_.field_6365_a == this.field_1212_e.field_6322_g.field_620_ab) {
-         object = this.field_1212_e.field_6322_g;
+         entity = this.field_1212_e.field_6322_g;
       }
 
-      if (object != null) {
-         ((Entity)object).func_6377_h(entity);
+      if (entity != null) {
+         entity.func_6377_h(entity1);
       }
    }
 
@@ -425,7 +_,7 @@
    }
 
    private Entity func_12246_a(int p_12246_1_) {
-      return (Entity)(p_12246_1_ == this.field_1212_e.field_6322_g.field_620_ab ? this.field_1212_e.field_6322_g : this.field_1211_f.func_709_b(p_12246_1_));
+      return p_12246_1_ == this.field_1212_e.field_6322_g.field_620_ab ? this.field_1212_e.field_6322_g : this.field_1211_f.func_709_b(p_12246_1_);
    }
 
    public void func_9446_a(Packet8 p_9446_1_) {
@@ -437,7 +_,7 @@
    }
 
    public void func_12245_a(Packet60 p_12245_1_) {
-      Explosion explosion = new Explosion(this.field_1212_e.field_6324_e, (Entity)null, p_12245_1_.field_12236_a, p_12245_1_.field_12235_b, p_12245_1_.field_12239_c, p_12245_1_.field_12238_d);
+      Explosion explosion = new Explosion(this.field_1212_e.field_6324_e, null, p_12245_1_.field_12236_a, p_12245_1_.field_12235_b, p_12245_1_.field_12239_c, p_12245_1_.field_12238_d);
       explosion.field_12251_g = p_12245_1_.field_12237_e;
       explosion.func_12247_b();
    }
