--- a/net/minecraft/src/PanelCrashReport.java
+++ b/net/minecraft/src/PanelCrashReport.java
@@ -75,7 +_,7 @@
       s3 = s3 + "\n";
       s3 = s3 + "\n";
       TextArea textarea = new TextArea(s3, 0, 0, 1);
-      textarea.setFont(new Font("Monospaced", 0, 12));
+      textarea.setFont(new Font("Monospaced", Font.PLAIN, 12));
       this.add(new CanvasMojangLogo(), "North");
       this.add(new CanvasCrashReport(80), "East");
       this.add(new CanvasCrashReport(80), "West");
