--- a/net/minecraft/src/RenderLiving.java
+++ b/net/minecraft/src/RenderLiving.java
@@ -2,13 +_,18 @@
 
 import org.lwjgl.opengl.GL11;
 
-public class RenderLiving extends Render {
+public class RenderLiving extends Render<EntityLiving> {
    protected ModelBase field_203_d;
    protected ModelBase field_6332_f;
 
    public RenderLiving(ModelBase p_i84_1_, float p_i84_2_) {
       this.field_203_d = p_i84_1_;
       this.field_9246_c = p_i84_2_;
+   }
+
+   @Override
+   public void func_147_a(EntityLiving p_147_1_, double p_147_2_, double p_147_4_, double p_147_6_, float p_147_8_, float p_147_9_) {
+      func_171_a(p_147_1_, p_147_2_, p_147_4_, p_147_6_, p_147_8_, p_147_9_);
    }
 
    public void func_4013_a(ModelBase p_4013_1_) {
