--- a/net/minecraft/src/RenderArrow.java
+++ b/net/minecraft/src/RenderArrow.java
@@ -2,7 +_,12 @@
 
 import org.lwjgl.opengl.GL11;
 
-public class RenderArrow extends Render {
+public class RenderArrow extends Render<EntityArrow> {
+   @Override
+   public void func_147_a(EntityArrow p_147_1_, double p_147_2_, double p_147_4_, double p_147_6_, float p_147_8_, float p_147_9_) {
+      func_154_a(p_147_1_, p_147_2_, p_147_4_, p_147_6_, p_147_8_, p_147_9_);
+   }
+
    public void func_154_a(EntityArrow p_154_1_, double p_154_2_, double p_154_4_, double p_154_6_, float p_154_8_, float p_154_9_) {
       this.func_151_a("/item/arrows.png");
       GL11.glPushMatrix();
@@ -32,27 +_,27 @@
       GL11.glTranslatef(-4.0F, 0.0F, 0.0F);
       GL11.glNormal3f(f8, 0.0F, 0.0F);
       tessellator.func_977_b();
-      tessellator.func_983_a(-7.0D, -2.0D, -2.0D, (double)f4, (double)f6);
-      tessellator.func_983_a(-7.0D, -2.0D, 2.0D, (double)f5, (double)f6);
-      tessellator.func_983_a(-7.0D, 2.0D, 2.0D, (double)f5, (double)f7);
-      tessellator.func_983_a(-7.0D, 2.0D, -2.0D, (double)f4, (double)f7);
+      tessellator.func_983_a(-7.0D, -2.0D, -2.0D, f4, f6);
+      tessellator.func_983_a(-7.0D, -2.0D, 2.0D, f5, f6);
+      tessellator.func_983_a(-7.0D, 2.0D, 2.0D, f5, f7);
+      tessellator.func_983_a(-7.0D, 2.0D, -2.0D, f4, f7);
       tessellator.func_982_a();
       GL11.glNormal3f(-f8, 0.0F, 0.0F);
       tessellator.func_977_b();
-      tessellator.func_983_a(-7.0D, 2.0D, -2.0D, (double)f4, (double)f6);
-      tessellator.func_983_a(-7.0D, 2.0D, 2.0D, (double)f5, (double)f6);
-      tessellator.func_983_a(-7.0D, -2.0D, 2.0D, (double)f5, (double)f7);
-      tessellator.func_983_a(-7.0D, -2.0D, -2.0D, (double)f4, (double)f7);
+      tessellator.func_983_a(-7.0D, 2.0D, -2.0D, f4, f6);
+      tessellator.func_983_a(-7.0D, 2.0D, 2.0D, f5, f6);
+      tessellator.func_983_a(-7.0D, -2.0D, 2.0D, f5, f7);
+      tessellator.func_983_a(-7.0D, -2.0D, -2.0D, f4, f7);
       tessellator.func_982_a();
 
       for(int i = 0; i < 4; ++i) {
          GL11.glRotatef(90.0F, 1.0F, 0.0F, 0.0F);
          GL11.glNormal3f(0.0F, 0.0F, f8);
          tessellator.func_977_b();
-         tessellator.func_983_a(-8.0D, -2.0D, 0.0D, (double)f, (double)f2);
-         tessellator.func_983_a(8.0D, -2.0D, 0.0D, (double)f1, (double)f2);
-         tessellator.func_983_a(8.0D, 2.0D, 0.0D, (double)f1, (double)f3);
-         tessellator.func_983_a(-8.0D, 2.0D, 0.0D, (double)f, (double)f3);
+         tessellator.func_983_a(-8.0D, -2.0D, 0.0D, f, f2);
+         tessellator.func_983_a(8.0D, -2.0D, 0.0D, f1, f2);
+         tessellator.func_983_a(8.0D, 2.0D, 0.0D, f1, f3);
+         tessellator.func_983_a(-8.0D, 2.0D, 0.0D, f, f3);
          tessellator.func_982_a();
       }
 
