--- a/net/minecraft/src/RenderFish.java
+++ b/net/minecraft/src/RenderFish.java
@@ -2,7 +_,13 @@
 
 import org.lwjgl.opengl.GL11;
 
-public class RenderFish extends Render {
+public class RenderFish extends Render<EntityFish> {
+
+   @Override
+   public void func_147_a(EntityFish p_147_1_, double p_147_2_, double p_147_4_, double p_147_6_, float p_147_8_, float p_147_9_) {
+      func_4011_a(p_147_1_, p_147_2_, p_147_4_, p_147_6_, p_147_8_, p_147_9_);
+   }
+
    public void func_4011_a(EntityFish p_4011_1_, double p_4011_2_, double p_4011_4_, double p_4011_6_, float p_4011_8_, float p_4011_9_) {
       GL11.glPushMatrix();
       GL11.glTranslatef((float)p_4011_2_, (float)p_4011_4_, (float)p_4011_6_);
@@ -23,27 +_,27 @@
       GL11.glRotatef(-this.field_191_a.field_1224_j, 1.0F, 0.0F, 0.0F);
       tessellator.func_977_b();
       tessellator.func_980_b(0.0F, 1.0F, 0.0F);
-      tessellator.func_983_a((double)(0.0F - f5), (double)(0.0F - f6), 0.0D, (double)f, (double)f3);
-      tessellator.func_983_a((double)(f4 - f5), (double)(0.0F - f6), 0.0D, (double)f1, (double)f3);
-      tessellator.func_983_a((double)(f4 - f5), (double)(1.0F - f6), 0.0D, (double)f1, (double)f2);
-      tessellator.func_983_a((double)(0.0F - f5), (double)(1.0F - f6), 0.0D, (double)f, (double)f2);
+      tessellator.func_983_a(0.0F - f5, 0.0F - f6, 0.0D, f, f3);
+      tessellator.func_983_a(f4 - f5, 0.0F - f6, 0.0D, f1, f3);
+      tessellator.func_983_a(f4 - f5, 1.0F - f6, 0.0D, f1, f2);
+      tessellator.func_983_a(0.0F - f5, 1.0F - f6, 0.0D, f, f2);
       tessellator.func_982_a();
       GL11.glDisable(32826);
       GL11.glPopMatrix();
       if (p_4011_1_.field_4097_b != null) {
          float f7 = (p_4011_1_.field_4097_b.field_603_as + (p_4011_1_.field_4097_b.field_605_aq - p_4011_1_.field_4097_b.field_603_as) * p_4011_9_) * (float)Math.PI / 180.0F;
          float f8 = (p_4011_1_.field_4097_b.field_602_at + (p_4011_1_.field_4097_b.field_604_ar - p_4011_1_.field_4097_b.field_602_at) * p_4011_9_) * (float)Math.PI / 180.0F;
-         double d0 = (double)MathHelper.func_1106_a(f7);
-         double d1 = (double)MathHelper.func_1114_b(f7);
-         double d2 = (double)MathHelper.func_1106_a(f8);
-         double d3 = (double)MathHelper.func_1114_b(f8);
+         double d0 = MathHelper.func_1106_a(f7);
+         double d1 = MathHelper.func_1114_b(f7);
+         double d2 = MathHelper.func_1106_a(f8);
+         double d3 = MathHelper.func_1114_b(f8);
          double d4 = p_4011_1_.field_4097_b.field_9285_at + (p_4011_1_.field_4097_b.field_611_ak - p_4011_1_.field_4097_b.field_9285_at) * (double)p_4011_9_ - d1 * 0.7D - d0 * 0.5D * d3;
          double d5 = p_4011_1_.field_4097_b.field_9284_au + (p_4011_1_.field_4097_b.field_610_al - p_4011_1_.field_4097_b.field_9284_au) * (double)p_4011_9_ - d2 * 0.5D;
          double d6 = p_4011_1_.field_4097_b.field_9283_av + (p_4011_1_.field_4097_b.field_609_am - p_4011_1_.field_4097_b.field_9283_av) * (double)p_4011_9_ - d0 * 0.7D + d1 * 0.5D * d3;
          if (this.field_191_a.field_1223_k.field_1560_x) {
             f7 = (p_4011_1_.field_4097_b.field_734_o + (p_4011_1_.field_4097_b.field_735_n - p_4011_1_.field_4097_b.field_734_o) * p_4011_9_) * (float)Math.PI / 180.0F;
-            d0 = (double)MathHelper.func_1106_a(f7);
-            d1 = (double)MathHelper.func_1114_b(f7);
+            d0 = MathHelper.func_1106_a(f7);
+            d1 = MathHelper.func_1114_b(f7);
             d4 = p_4011_1_.field_4097_b.field_9285_at + (p_4011_1_.field_4097_b.field_611_ak - p_4011_1_.field_4097_b.field_9285_at) * (double)p_4011_9_ - d1 * 0.35D - d0 * 0.85D;
             d5 = p_4011_1_.field_4097_b.field_9284_au + (p_4011_1_.field_4097_b.field_610_al - p_4011_1_.field_4097_b.field_9284_au) * (double)p_4011_9_ - 0.45D;
             d6 = p_4011_1_.field_4097_b.field_9283_av + (p_4011_1_.field_4097_b.field_609_am - p_4011_1_.field_4097_b.field_9283_av) * (double)p_4011_9_ - d0 * 0.35D + d1 * 0.85D;
@@ -52,9 +_,9 @@
          double d7 = p_4011_1_.field_9285_at + (p_4011_1_.field_611_ak - p_4011_1_.field_9285_at) * (double)p_4011_9_;
          double d8 = p_4011_1_.field_9284_au + (p_4011_1_.field_610_al - p_4011_1_.field_9284_au) * (double)p_4011_9_ + 0.25D;
          double d9 = p_4011_1_.field_9283_av + (p_4011_1_.field_609_am - p_4011_1_.field_9283_av) * (double)p_4011_9_;
-         double d10 = (double)((float)(d4 - d7));
-         double d11 = (double)((float)(d5 - d8));
-         double d12 = (double)((float)(d6 - d9));
+         double d10 = (float)(d4 - d7);
+         double d11 = (float)(d5 - d8);
+         double d12 = (float)(d6 - d9);
          GL11.glDisable(3553);
          GL11.glDisable(2896);
          tessellator.func_992_a(3);
