--- a/net/minecraft/src/RenderGlobal.java
+++ b/net/minecraft/src/RenderGlobal.java
@@ -12,10 +12,10 @@
 import org.lwjgl.opengl.GL11;
 
 public class RenderGlobal implements IWorldAccess {
-   public List field_1458_a = new ArrayList();
+   public List<TileEntity> field_1458_a = new ArrayList<>();
    private World field_1448_k;
    private RenderEngine field_1447_l;
-   private List field_1446_m = new ArrayList();
+   private List<WorldRenderer> field_1446_m = new ArrayList<>();
    private WorldRenderer[] field_1445_n;
    private WorldRenderer[] field_1444_o;
    private int field_1443_p;
@@ -48,7 +48,7 @@
    private int field_1418_O;
    private int field_1417_P;
    private int field_1416_Q;
-   private List field_1415_R = new ArrayList();
+   private List<WorldRenderer> field_1415_R = new ArrayList<>();
    private RenderList[] field_1414_S = new RenderList[]{new RenderList(), new RenderList(), new RenderList(), new RenderList()};
    int field_1455_d = 0;
    int field_1454_e = GLAllocation.func_1124_a(1);
@@ -65,11 +65,11 @@
       this.field_1440_s = GLAllocation.func_1124_a(b0 * b0 * b0 * 3);
       this.field_1436_w = p_i101_1_.func_6251_l().func_1213_a();
       if (this.field_1436_w) {
-         ((Buffer)this.field_1456_c).clear();
+         this.field_1456_c.clear();
          this.field_1437_v = GLAllocation.func_1125_c(b0 * b0 * b0);
-         ((Buffer)this.field_1437_v).clear();
-         ((Buffer)this.field_1437_v).position(0);
-         ((Buffer)this.field_1437_v).limit(b0 * b0 * b0);
+         this.field_1437_v.clear();
+         this.field_1437_v.position(0);
+         this.field_1437_v.limit(b0 * b0 * b0);
          ARBOcclusionQuery.glGenQueriesARB(this.field_1437_v);
       }
 
@@ -839,7 +839,7 @@
       int j = this.field_1446_m.size();
 
       for(int k = 0; k < j; ++k) {
-         WorldRenderer worldrenderer = (WorldRenderer)this.field_1446_m.get(i - k);
+         WorldRenderer worldrenderer = this.field_1446_m.get(i - k);
          if (!p_948_2_) {
             if (worldrenderer.func_1202_a(p_948_1_) > 1024.0F) {
                if (worldrenderer.field_1749_o) {
@@ -952,7 +952,7 @@
             double d0 = p_955_1_.field_638_aI + (p_955_1_.field_611_ak - p_955_1_.field_638_aI) * (double)p_955_5_;
             double d1 = p_955_1_.field_637_aJ + (p_955_1_.field_610_al - p_955_1_.field_637_aJ) * (double)p_955_5_;
             double d2 = p_955_1_.field_636_aK + (p_955_1_.field_609_am - p_955_1_.field_636_aK) * (double)p_955_5_;
-            this.func_942_a(Block.field_345_n[i].func_246_f(this.field_1448_k, p_955_2_.field_1166_b, p_955_2_.field_1172_c, p_955_2_.field_1171_d).func_1177_b((double)f, (double)f, (double)f).func_1166_c(-d0, -d1, -d2));
+            this.func_942_a(Block.field_345_n[i].func_246_f(this.field_1448_k, p_955_2_.field_1166_b, p_955_2_.field_1172_c, p_955_2_.field_1171_d).func_1177_b(f, f, f).func_1166_c(-d0, -d1, -d2));
          }
 
          GL11.glDepthMask(true);
