--- a/net/minecraft/src/RenderHelper.java
+++ b/net/minecraft/src/RenderHelper.java
@@ -1,6 +_,5 @@
 package net.minecraft.src;
 
-import java.nio.Buffer;
 import java.nio.FloatBuffer;
 import org.lwjgl.opengl.GL11;
 
@@ -23,12 +_,12 @@
       float f = 0.4F;
       float f1 = 0.6F;
       float f2 = 0.0F;
-      Vec3D vec3d = Vec3D.func_1248_b((double)0.2F, 1.0D, (double)-0.7F).func_1252_b();
+      Vec3D vec3d = Vec3D.func_1248_b(0.2F, 1.0D, -0.7F).func_1252_b();
       GL11.glLight(16384, 4611, func_1157_a(vec3d.field_1776_a, vec3d.field_1775_b, vec3d.field_1779_c, 0.0D));
       GL11.glLight(16384, 4609, func_1156_a(f1, f1, f1, 1.0F));
       GL11.glLight(16384, 4608, func_1156_a(0.0F, 0.0F, 0.0F, 1.0F));
       GL11.glLight(16384, 4610, func_1156_a(f2, f2, f2, 1.0F));
-      vec3d = Vec3D.func_1248_b((double)-0.2F, 1.0D, (double)0.7F).func_1252_b();
+      vec3d = Vec3D.func_1248_b(-0.2F, 1.0D, 0.7F).func_1252_b();
       GL11.glLight(16385, 4611, func_1157_a(vec3d.field_1776_a, vec3d.field_1775_b, vec3d.field_1779_c, 0.0D));
       GL11.glLight(16385, 4609, func_1156_a(f1, f1, f1, 1.0F));
       GL11.glLight(16385, 4608, func_1156_a(0.0F, 0.0F, 0.0F, 1.0F));
@@ -42,9 +_,9 @@
    }
 
    private static FloatBuffer func_1156_a(float p_1156_0_, float p_1156_1_, float p_1156_2_, float p_1156_3_) {
-      ((Buffer)field_1695_a).clear();
+      field_1695_a.clear();
       field_1695_a.put(p_1156_0_).put(p_1156_1_).put(p_1156_2_).put(p_1156_3_);
-      ((Buffer)field_1695_a).flip();
+      field_1695_a.flip();
       return field_1695_a;
    }
 }
