--- a/net/minecraft/src/RenderPlayer.java
+++ b/net/minecraft/src/RenderPlayer.java
@@ -52,7 +52,7 @@
       float f2 = p_188_1_.func_379_d(this.field_191_a.field_1226_h);
       float f3 = p_188_1_.func_381_o() ? 32.0F : 64.0F;
       if (f2 < f3) {
-         f1 = (float)((double)f1 * (Math.sqrt((double)f2) / 2.0D));
+         f1 = (float)((double)f1 * (Math.sqrt(f2) / 2.0D));
          FontRenderer fontrenderer = this.func_6329_a();
          GL11.glPushMatrix();
          GL11.glTranslatef((float)p_188_2_ + 0.0F, (float)p_188_4_ + 2.3F, (float)p_188_6_);
@@ -72,20 +72,16 @@
             tessellator.func_977_b();
             int i = fontrenderer.func_871_a(s) / 2;
             tessellator.func_986_a(0.0F, 0.0F, 0.0F, 0.25F);
-            tessellator.func_991_a((double)(-i - 1), -1.0D, 0.0D);
-            tessellator.func_991_a((double)(-i - 1), 8.0D, 0.0D);
-            tessellator.func_991_a((double)(i + 1), 8.0D, 0.0D);
-            tessellator.func_991_a((double)(i + 1), -1.0D, 0.0D);
+            tessellator.func_991_a(-i - 1, -1.0D, 0.0D);
+            tessellator.func_991_a(-i - 1, 8.0D, 0.0D);
+            tessellator.func_991_a(i + 1, 8.0D, 0.0D);
+            tessellator.func_991_a(i + 1, -1.0D, 0.0D);
             tessellator.func_982_a();
             GL11.glEnable(3553);
             fontrenderer.func_873_b(s, -fontrenderer.func_871_a(s) / 2, 0, 553648127);
             GL11.glEnable(2929);
             GL11.glDepthMask(true);
             fontrenderer.func_873_b(s, -fontrenderer.func_871_a(s) / 2, 0, -1);
-            GL11.glEnable(2896);
-            GL11.glDisable(3042);
-            GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-            GL11.glPopMatrix();
          } else {
             GL11.glTranslatef(0.0F, 0.25F / f1, 0.0F);
             GL11.glDepthMask(false);
@@ -96,20 +92,20 @@
             tessellator1.func_977_b();
             int j = fontrenderer.func_871_a(s) / 2;
             tessellator1.func_986_a(0.0F, 0.0F, 0.0F, 0.25F);
-            tessellator1.func_991_a((double)(-j - 1), -1.0D, 0.0D);
-            tessellator1.func_991_a((double)(-j - 1), 8.0D, 0.0D);
-            tessellator1.func_991_a((double)(j + 1), 8.0D, 0.0D);
-            tessellator1.func_991_a((double)(j + 1), -1.0D, 0.0D);
+            tessellator1.func_991_a(-j - 1, -1.0D, 0.0D);
+            tessellator1.func_991_a(-j - 1, 8.0D, 0.0D);
+            tessellator1.func_991_a(j + 1, 8.0D, 0.0D);
+            tessellator1.func_991_a(j + 1, -1.0D, 0.0D);
             tessellator1.func_982_a();
             GL11.glEnable(3553);
             GL11.glDepthMask(true);
             fontrenderer.func_873_b(s, -fontrenderer.func_871_a(s) / 2, 0, 553648127);
+         }
             GL11.glEnable(2896);
             GL11.glDisable(3042);
             GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
             GL11.glPopMatrix();
          }
-      }
 
    }
 
