--- a/net/minecraft/src/ScreenShotHelper.java
+++ b/net/minecraft/src/ScreenShotHelper.java
@@ -2,7 +2,6 @@
 
 import java.awt.image.BufferedImage;
 import java.io.File;
-import java.nio.Buffer;
 import java.nio.ByteBuffer;
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
@@ -29,9 +28,9 @@
 
          GL11.glPixelStorei(3333, 1);
          GL11.glPixelStorei(3317, 1);
-         ((Buffer)field_4286_b).clear();
+         field_4286_b.clear();
          GL11.glReadPixels(0, 0, p_4148_1_, p_4148_2_, 6407, 5121, field_4286_b);
-         ((Buffer)field_4286_b).clear();
+         field_4286_b.clear();
          String s = "" + field_4287_a.format(new Date());
 
          File file2;
