--- a/net/minecraft/src/TexturePackList.java
+++ b/net/minecraft/src/TexturePackList.java
@@ -1,6 +_,7 @@
 package net.minecraft.src;
 
 import java.io.File;
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
@@ -8,10 +_,10 @@
 import net.minecraft.client.Minecraft;
 
 public class TexturePackList {
-   private List field_6533_b = new ArrayList();
+   private List<TexturePackBase> field_6533_b = new ArrayList<>();
    private TexturePackBase field_6539_c = new TexturePackDefault();
    public TexturePackBase field_6534_a;
-   private Map field_6538_d = new HashMap();
+   private Map<String, TexturePackCustom> field_6538_d = new HashMap<>();
    private Minecraft field_6537_e;
    private File field_6536_f;
    private String field_6535_g;
@@ -43,10 +_,38 @@
    }
 
    public void func_6532_a() {
-      // $FF: Couldn't be decompiled
+      ArrayList<TexturePackBase> arrayList = new ArrayList<>();
+      this.field_6534_a = null;
+      arrayList.add(this.field_6539_c);
+      if (this.field_6536_f.exists() && this.field_6536_f.isDirectory()) {
+         File[] arrayOfFile = this.field_6536_f.listFiles();
+         for (File file : arrayOfFile) {
+            if (file.isFile() && file.getName().toLowerCase().endsWith(".zip")) {
+               String str = file.getName() + ":" + file.length() + ":" + file.lastModified();
+               if (!this.field_6538_d.containsKey(str)) {
+                  TexturePackCustom texturePackCustom = new TexturePackCustom(file);
+                  texturePackCustom.field_6488_d = str;
+                  this.field_6538_d.put(str, texturePackCustom);
+                  texturePackCustom.func_6485_a(this.field_6537_e);
+   }
+               TexturePackBase texturePackBase = this.field_6538_d.get(str);
+               if (texturePackBase.field_6487_a.equals(this.field_6535_g))
+                  this.field_6534_a = texturePackBase;
+               arrayList.add(texturePackBase);
+            }
+         }
+      }
+      if (this.field_6534_a == null)
+         this.field_6534_a = this.field_6539_c;
+      this.field_6533_b.removeAll(arrayList);
+      for (TexturePackBase texturePackBase : this.field_6533_b) {
+         texturePackBase.func_6484_b(this.field_6537_e);
+         this.field_6538_d.remove(texturePackBase.field_6488_d);
+      }
+      this.field_6533_b = arrayList;
    }
 
-   public List func_6530_b() {
-      return new ArrayList(this.field_6533_b);
+   public List<TexturePackBase> func_6530_b() {
+      return new ArrayList<>(this.field_6533_b);
    }
 }
