--- a/net/minecraft/src/ThreadDownloadResources.java
+++ b/net/minecraft/src/ThreadDownloadResources.java
@@ -42,8 +_,8 @@
                Node node = nodelist.item(j);
                if (node.getNodeType() == 1) {
                   Element element = (Element)node;
-                  String s = ((Element)element.getElementsByTagName("Key").item(0)).getChildNodes().item(0).getNodeValue();
-                  long k = Long.parseLong(((Element)element.getElementsByTagName("Size").item(0)).getChildNodes().item(0).getNodeValue());
+                  String s = element.getElementsByTagName("Key").item(0).getChildNodes().item(0).getNodeValue();
+                  long k = Long.parseLong(element.getElementsByTagName("Size").item(0).getChildNodes().item(0).getNodeValue());
                   if (k > 0L) {
                      this.func_1211_a(url, s, k, i);
                      if (this.field_1769_c) {
@@ -67,14 +_,14 @@
    private void func_1209_a(File p_1209_1_, String p_1209_2_) {
       File[] afile = p_1209_1_.listFiles();
 
-      for(int i = 0; i < afile.length; ++i) {
-         if (afile[i].isDirectory()) {
-            this.func_1209_a(afile[i], p_1209_2_ + afile[i].getName() + "/");
+      for (File file : afile) {
+         if (file.isDirectory()) {
+            this.func_1209_a(file, p_1209_2_ + file.getName() + "/");
          } else {
             try {
-               this.field_1767_b.func_6268_a(p_1209_2_ + afile[i].getName(), afile[i]);
+               this.field_1767_b.func_6268_a(p_1209_2_ + file.getName(), file);
             } catch (Exception exception) {
-               System.out.println("Failed to add " + p_1209_2_ + afile[i].getName());
+               System.out.println("Failed to add " + p_1209_2_ + file.getName());
             }
          }
       }
