--- a/net/minecraft/src/ThreadDownloadImage.java
+++ b/net/minecraft/src/ThreadDownloadImage.java
@@ -33,12 +_,9 @@
             } else {
                this.field_1217_c.field_1706_a = this.field_1215_b.func_883_a(ImageIO.read(httpurlconnection.getInputStream()));
             }
-
-            return;
          }
       } catch (Exception exception) {
          exception.printStackTrace();
-         return;
       } finally {
          httpurlconnection.disconnect();
       }
