--- a/net/minecraft/src/WorldClient.java
+++ b/net/minecraft/src/WorldClient.java
@@ -6,13 +_,13 @@
 import java.util.Set;
 
 public class WorldClient extends World {
-   private LinkedList field_1057_z = new LinkedList();
+   private LinkedList<WorldBlockPositionType> field_1057_z = new LinkedList<>();
    private NetClientHandler field_1052_A;
    private ChunkProviderClient field_1058_B;
    private boolean field_1056_C = false;
    private MCHashTable field_1055_D = new MCHashTable();
-   private Set field_1054_E = new HashSet();
-   private Set field_1053_F = new HashSet();
+   private Set<Entity> field_1054_E = new HashSet<>();
+   private Set<Entity> field_1053_F = new HashSet<>();
 
    public WorldClient(NetClientHandler p_i158_1_, long p_i158_2_, int p_i158_4_) {
       super("MpServer", WorldProvider.func_4101_a(p_i158_4_), p_i158_2_);
@@ -28,13 +_,13 @@
       if (i != this.field_1046_e) {
          this.field_1046_e = i;
 
-         for(int j = 0; j < this.field_9434_r.size(); ++j) {
-            ((IWorldAccess)this.field_9434_r.get(j)).func_936_e();
+         for (IWorldAccess iWorldAccess : this.field_9434_r) {
+            iWorldAccess.func_936_e();
          }
       }
 
       for(int k = 0; k < 10 && !this.field_1053_F.isEmpty(); ++k) {
-         Entity entity = (Entity)this.field_1053_F.iterator().next();
+         Entity entity = this.field_1053_F.iterator().next();
          if (!this.field_1050_a.contains(entity)) {
             this.func_674_a(entity);
          }
@@ -43,7 +_,7 @@
       this.field_1052_A.func_848_a();
 
       for(int l = 0; l < this.field_1057_z.size(); ++l) {
-         WorldBlockPositionType worldblockpositiontype = (WorldBlockPositionType)this.field_1057_z.get(l);
+         WorldBlockPositionType worldblockpositiontype = this.field_1057_z.get(l);
          if (--worldblockpositiontype.field_1206_d == 0) {
             super.func_643_a(worldblockpositiontype.field_1202_a, worldblockpositiontype.field_1201_b, worldblockpositiontype.field_1207_c, worldblockpositiontype.field_1205_e, worldblockpositiontype.field_1204_f);
             super.func_665_h(worldblockpositiontype.field_1202_a, worldblockpositiontype.field_1201_b, worldblockpositiontype.field_1207_c);
@@ -55,7 +_,7 @@
 
    public void func_711_c(int p_711_1_, int p_711_2_, int p_711_3_, int p_711_4_, int p_711_5_, int p_711_6_) {
       for(int i = 0; i < this.field_1057_z.size(); ++i) {
-         WorldBlockPositionType worldblockpositiontype = (WorldBlockPositionType)this.field_1057_z.get(i);
+         WorldBlockPositionType worldblockpositiontype = this.field_1057_z.get(i);
          if (worldblockpositiontype.field_1202_a >= p_711_1_ && worldblockpositiontype.field_1201_b >= p_711_2_ && worldblockpositiontype.field_1207_c >= p_711_3_ && worldblockpositiontype.field_1202_a <= p_711_4_ && worldblockpositiontype.field_1201_b <= p_711_5_ && worldblockpositiontype.field_1207_c <= p_711_6_) {
             this.field_1057_z.remove(i--);
          }
