--- a/net/minecraft/src/WorldGenFlowers.java
+++ b/net/minecraft/src/WorldGenFlowers.java
@@ -14,7 +14,7 @@
          int j = p_516_3_ + p_516_2_.nextInt(8) - p_516_2_.nextInt(8);
          int k = p_516_4_ + p_516_2_.nextInt(4) - p_516_2_.nextInt(4);
          int l = p_516_5_ + p_516_2_.nextInt(8) - p_516_2_.nextInt(8);
-         if (p_516_1_.func_600_a(j, k, l) == 0 && ((BlockFlower)Block.field_345_n[this.field_885_a]).func_223_g(p_516_1_, j, k, l)) {
+         if (p_516_1_.func_600_a(j, k, l) == 0 && Block.field_345_n[this.field_885_a].func_223_g(p_516_1_, j, k, l)) {
             p_516_1_.func_634_a(j, k, l, this.field_885_a);
          }
       }
